<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="OmnipathR">
<title>OmnipathR: an R client for the OmniPath web service • OmnipathR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="../deps/bs3compat-0.2.5.1/tabs.js"></script><script src="../deps/bs3compat-0.2.5.1/bs3compat.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="OmnipathR: an R client for the OmniPath web service">
<meta property="og:description" content="OmnipathR">
<meta property="og:image" content="https://saezlab.github.io/OmnipathR/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-10"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-10');
</script>
</head>
<body data-spy="scroll" data-target="#toc" data-headroom>
    

      <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    <a class="navbar-brand mb-0 h1" href="../index.html">OmnipathR</a>

    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <button type="button" id="version-badge" class="badge badge-default d-none d-lg-block" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Released version">3.1.2</button>

    <div id="navbar" class="collapse navbar-collapse">
      <ul class="navbar-nav mr-auto ml-3">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bioc_workshop.html">OmniPath Bioconductor workshop</a>
    <a class="dropdown-item" href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    <a class="dropdown-item" href="../articles/gene_ontology.html">Gene Ontology in OmnipathR</a>
    <a class="dropdown-item" href="../articles/nichenet.html">Using NicheNet with OmnipathR</a>
    <a class="dropdown-item" href="../articles/omnipath_intro.html">OmnipathR: an R client for the OmniPath web service</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control mr-sm-2" aria-label="Search" name="search-input" data-search-index="/search.json" id="search-input" placeholder="Search..." autocomplete="off">
</form>

      <ul class="navbar-nav ml-1">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/OmnipathR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/omnipathdb">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://omnipathdb.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
  </div>
</nav><div class="container template-article">
<script src="omnipath_intro_files/header-attrs-2.9.5/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <img src="../package-logo.png" class="pkg-logo" alt=""><div class="page-header pb-2 mt-4 mb-2 border-bottom toc-ignore">
      <h1 class="display-4" data-toc-skip>OmnipathR: an R client for the OmniPath web service</h1>
                        <h4 data-toc-skip class="author">Alberto Valdeolivas</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:alvaldeolivas@gmail.com" class="email">alvaldeolivas@gmail.com</a>
      
                              <h4 data-toc-skip class="author">Attila Gabor</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:attila.gabor@bioquant.uni-heidelberg.de" class="email">attila.gabor@bioquant.uni-heidelberg.de</a>
      
                              <h4 data-toc-skip class="author">Denes Turei</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:turei.denes@gmail.com" class="email">turei.denes@gmail.com</a>
      
                              <h4 data-toc-skip class="author">Julio Saez-Rodriguez</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/OmnipathR/blob/master/vignettes/omnipath_intro.Rmd" class="external-link"><code>vignettes/omnipath_intro.Rmd</code></a></small>
      <div class="d-none name"><code>omnipath_intro.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes how to use the <em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> package to retrieve information from the OmniPath database: <a href="https://omnipathdb.org/" class="external-link uri">https://omnipathdb.org/</a>. In addition, it includes some utility functions to filter analyse and visualize the data.</p>
    </div>
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor" aria-hidden="true"></a>Introduction</h1>
<p><em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> is an <em>R</em> package built to provide easy access to the data stored in the OmniPath webservice <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>:</p>
<p><a href="http://omnipathdb.org/" class="external-link uri">http://omnipathdb.org/</a></p>
<p>The webservice implements a very simple REST style API. This package make requests by the HTTP protocol to retreive the data. Hence, fast Internet access is required for a proper use of <em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em>.</p>
<div id="query-types" class="section level2">
<h2 class="hasAnchor">
<a href="#query-types" class="anchor" aria-hidden="true"></a>Query types</h2>
<p><em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> can retrieve five different types of data:</p>
<ul>
<li>
<p><strong>Interactions</strong>: protein-protein interactions organized in different datasets:</p>
<ul>
<li>
<strong>omnipath</strong>: the OmniPath data as defined in the original publication <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span> and collected from different databases.</li>
<li>
<strong>pathwayextra</strong>: activity flow interactions without literature reference.</li>
<li>
<strong>kinaseextra</strong>: enzyme-substrate interactions without literature reference.</li>
<li>
<strong>ligrecextra</strong>: ligand-receptor interactions without literature reference.</li>
<li>
<strong>tfregulons</strong>: transcription factor (TF)-target interactions from <strong>DoRothEA</strong> <span class="citation">(Garcia-Alonso et al. 2019)</span>.</li>
<li>
<strong>tf-miRNA</strong>: transcription factor-miRNA interactions</li>
<li>
<strong>miRNA-target</strong>: miRNA-mRNA interactions.</li>
<li>
<strong>lncRNA-mRNA</strong>: lncRNA-mRNA interactions.</li>
</ul>
</li>
<li><p><strong>Post-translational modifications (PTMs)</strong>: It provides enzyme-substrate reactions in a very similar way to the aforementioned interactions. Some of the biological databases related to PTMs integrated in OmniPath are Phospho.ELM <span class="citation">(Dinkel et al. 2010)</span> and PhosphoSitePlus [<span class="citation">Hornbeck et al. (2014)</span>}.</p></li>
<li><p><strong>Complexes</strong>: it provides access to a comprehensive database of more than 22000 protein complexes. This data comes from different resources such as: CORUM <span class="citation">(Giurgiu et al. 2018)</span> or Hu.map <span class="citation">(Drew et al. 2017)</span>.</p></li>
<li><p><strong>Annotations</strong>: it provides a large variety of data regarding different annotations about proteins and complexes. These data come from dozens of databases covering different topics such as: The Topology Data Bank of Transmembrane Proteins (TOPDB) <span class="citation">(Dobson et al. 2014)</span> or ExoCarta <span class="citation">(Keerthikumar et al. 2016)</span>, a database collecting the proteins that were identified in exosomes in multiple organisms.</p></li>
<li><p><strong>Intercell</strong>: it provides information on the roles in inter-cellular signaling. For instance. if a protein is a ligand, a receptor, an extracellular matrix (ECM) component, etc. The data does not come from original sources but combined from several databases by us. The source databases, such as CellPhoneDB <span class="citation">(Vento-Tormo et al. 2018)</span> or Receptome <span class="citation">(Ben-Shlomo et al. 2003)</span>, are also referred for each reacord.</p></li>
</ul>
<p>Figure @ref(fig:fig1) shows an overview of the resources featured in <strong>OmniPath</strong>. For more detailed information about the original data sources integrated in Omnipath, please visit:</p>
<ul>
<li><a href="http://omnipathdb.org/" class="external-link uri">http://omnipathdb.org/</a></li>
<li><a href="http://omnipathdb.org/info" class="external-link uri">http://omnipathdb.org/info</a></li>
</ul>
<p><code>{r, fig1, dpi=300, fig.width=10, fig.height=10, fig.cap="Overview of the resources featured in OmniPath. Causal resources (including activity-flow and enzyme-substrate resources) can provide direction (*) or sign and direction (+) of interactions.", echo=FALSE} library(knitr) knitr::include_graphics("../man/figures/page1_1.png")</code></p>
</div>
<div id="mouse-and-rat" class="section level2">
<h2 class="hasAnchor">
<a href="#mouse-and-rat" class="anchor" aria-hidden="true"></a>Mouse and rat</h2>
<p>Excluding the miRNA interactions, all interactions and PTMs are available for human, mouse and rat. The rodent data has been translated from human using the NCBI Homologene database. Many human proteins do not have known homolog in rodents, hence rodent datasets are smaller than their human counterparts.</p>
<p>In case you work with mouse omics data you might do better to translate your dataset to human (for example using the pypath.homology module, <a href="https://github.com/saezlab/pypath/" class="external-link uri">https://github.com/saezlab/pypath/</a>) and use human interaction data.</p>
</div>
</div>
<div id="installation-of-the-omnipathr-package" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-of-the-omnipathr-package" class="anchor" aria-hidden="true"></a>Installation of the <em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> package</h1>
<p>First of all, you need a current version of <em>R</em>. <em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> is a freely available package deposited on <em>Bioconductor</em> and <a href="https://github.com/saezlab/OmnipathR" class="external-link">GitHub</a>. You can install it by running the following commands on an <em>R</em> console:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"OmnipathR"</span><span class="op">)</span></code></pre></div>
<p>We also load here the required packages to run the code in this vignette.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/OmnipathR/">OmnipathR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dnet.r-forge.r-project.org" class="external-link">dnet</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gprofiler2</span><span class="op">)</span></code></pre></div>
</div>
<div id="usage-examples" class="section level1">
<h1 class="hasAnchor">
<a href="#usage-examples" class="anchor" aria-hidden="true"></a>Usage Examples</h1>
<p>In the following paragraphs, we provide some examples to describe how to use the <em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> package to retrieve different types of information from <strong>Omnipath</strong> webserver. In addition, we play around with the data aiming at obtaining some biological relevant information.</p>
<p>Noteworthy, the sections <strong>complexes</strong>, <strong>annotations</strong> and <strong>intercell</strong> are linked. We explore the annotations and roles in inter-cellular communications of the proteins involved in a given complex. This basic example shows the usefulness of integrating the information available in the different <strong>Omnipath</strong> resources.</p>
<div id="interactions" class="section level2">
<h2 class="hasAnchor">
<a href="#interactions" class="anchor" aria-hidden="true"></a>Interactions</h2>
<p>Proteins interact among them and with other biological molecules to perform cellular functions. Proteins also participates in pathways, linked series of reactions occurring inter/intra cells to transform products or to transmit signals inducing specific cellular responses. Protein interactions are therefore a very valuable source of information to understand cellular functioning.</p>
<p>We here download the original <strong>OmniPath</strong> human interactions <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>. To do so, we first check the different source databases and select some of them. Then, we print some of the downloaded interactions (“+” means activation, “-” means inhibition and “?” means undirected interactions or inconclusive data).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We check some of the different interaction databases</span>
<span class="fu"><a href="../reference/get_interaction_resources.html">get_interaction_resources</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##   [1] "ABS"                         "ACSN"                       
##   [3] "ACSN_SignaLink3"             "Adhesome"                   
##   [5] "AlzPathway"                  "ARACNe-GTEx_DoRothEA"       
##   [7] "ARN"                         "Baccin2019"                 
##   [9] "BEL-Large-Corpus_ProtMapper" "BioGRID"                    
##  [11] "BioGRID_ICELLNET"            "CA1"                        
##  [13] "CancerCellMap"               "CellChatDB"                 
##  [15] "CellChatDB-cofactors"        "CellPhoneDB"                
##  [17] "CellPhoneDB_ICELLNET"        "CellTalkDB"                 
##  [19] "connectomeDB2020"            "dbPTM"                      
##  [21] "DeathDomain"                 "DEPOD"                      
##  [23] "Dinarello2013_ICELLNET"      "DIP"                        
##  [25] "DLRP_talklr"                 "DOMINO"                     
##  [27] "DoRothEA"                    "DoRothEA-reviews_DoRothEA"  
##  [29] "ELM"                         "EMBRACE"                    
##  [31] "ENCODE_tf-mirna"             "ENCODE-distal"              
##  [33] "ENCODE-proximal"             "FANTOM4_DoRothEA"           
##  [35] "Fantom5_LRdb"                "GO-lig-rec_ICELLNET"        
##  [37] "Guide2Pharma"                "Guide2Pharma_CellPhoneDB"   
##  [39] "Guide2Pharma_ICELLNET"       "Guide2Pharma_LRdb"          
##  [41] "Guide2Pharma_talklr"         "HOCOMOCO_DoRothEA"          
##  [43] "HPMR"                        "HPMR_ICELLNET"              
##  [45] "HPMR_LRdb"                   "HPMR_talklr"                
##  [47] "HPRD"                        "HPRD_KEA"                   
##  [49] "HPRD_LRdb"                   "HPRD_MIMP"                  
##  [51] "HPRD_talklr"                 "HPRD-phos"                  
##  [53] "HTRIdb"                      "HTRIdb_DoRothEA"            
##  [55] "HuRI"                        "I2D_CellPhoneDB"            
##  [57] "ICELLNET"                    "IMEx_CellPhoneDB"           
##  [59] "InnateDB"                    "InnateDB_CellPhoneDB"       
##  [61] "InnateDB_ICELLNET"           "InnateDB_SignaLink3"        
##  [63] "InnateDB-All_CellPhoneDB"    "IntAct"                     
##  [65] "IntAct_CellPhoneDB"          "IntAct_DoRothEA"            
##  [67] "iPTMnet"                     "iTALK"                      
##  [69] "JASPAR_DoRothEA"             "KEA"                        
##  [71] "KEGG-MEDICUS"                "Kinexus_KEA"                
##  [73] "Kirouac2010"                 "Kirouac2010_ICELLNET"       
##  [75] "Li2012"                      "Lit-BM-17"                  
##  [77] "LMPID"                       "lncrnadb"                   
##  [79] "LncRNADisease"               "LRdb"                       
##  [81] "Macrophage"                  "Macrophage_ICELLNET"        
##  [83] "MatrixDB"                    "MatrixDB_CellPhoneDB"       
##  [85] "MIMP"                        "MINT_CellPhoneDB"           
##  [87] "miR2Disease"                 "miRDeathDB"                 
##  [89] "miRecords"                   "miRTarBase"                 
##  [91] "MPPI"                        "NCI-PID_ProtMapper"         
##  [93] "ncRDeathDB"                  "NetPath"                    
##  [95] "NetworKIN_KEA"               "NFIRegulomeDB_DoRothEA"     
##  [97] "NRF2ome"                     "ORegAnno"                   
##  [99] "ORegAnno_DoRothEA"           "PAZAR"                      
## [101] "PAZAR_DoRothEA"              "phosphoELM"                 
## [103] "phosphoELM_KEA"              "phosphoELM_MIMP"            
## [105] "PhosphoNetworks"             "PhosphoPoint"               
## [107] "PhosphoSite"                 "PhosphoSite_KEA"            
## [109] "PhosphoSite_MIMP"            "PhosphoSite_noref"          
## [111] "PhosphoSite_ProtMapper"      "ProtMapper"                 
## [113] "Ramilowski2015"              "Ramilowski2015_Baccin2019"  
## [115] "Ramilowski2015_ICELLNET"     "REACH_ProtMapper"           
## [117] "Reactome_ICELLNET"           "Reactome_LRdb"              
## [119] "Reactome_ProtMapper"         "Reactome_SignaLink3"        
## [121] "RegNetwork_DoRothEA"         "ReMap_DoRothEA"             
## [123] "RLIMS-P_ProtMapper"          "SignaLink3"                 
## [125] "SignaLink3_ICELLNET"         "SIGNOR"                     
## [127] "SIGNOR_ICELLNET"             "SIGNOR_ProtMapper"          
## [129] "Sparser_ProtMapper"          "SPIKE"                      
## [131] "SPIKE_ICELLNET"              "STRING_ICELLNET"            
## [133] "STRING_talklr"               "talklr"                     
## [135] "TCRcuration_SignaLink3"      "TFactS_DoRothEA"            
## [137] "TFe_DoRothEA"                "TransmiR"                   
## [139] "TRED_DoRothEA"               "TRIP"                       
## [141] "TRRD_DoRothEA"               "TRRUST_DoRothEA"            
## [143] "UniProt_CellPhoneDB"         "UniProt_LRdb"               
## [145] "Wang"                        "Wojtowicz2020"</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## The interactions are stored into a data frame.</span>
<span class="va">interactions</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions</a></span><span class="op">(</span>resources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SignaLink3"</span>,<span class="st">"PhosphoSite"</span>,
    <span class="st">"SIGNOR"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## We visualize the first interactions in the data frame.</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">interactions</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 5
##   source    interaction target    n_resources n_references                      
##   &lt;chr&gt;     &lt;chr&gt;       &lt;chr&gt;           &lt;int&gt; &lt;chr&gt;                             
## 1 RACK1 (P… ==( - )==&gt;  TRPM6 (Q…           2 SIGNOR:18258429;TRIP:18258429     
## 2 PRKG1 (Q… ==( - )==&gt;  TRPC6 (Q…           4 PhosphoSite:19961855;PhosphoSite:…
## 3 SRC (P12… ==( + )==&gt;  TRPV1 (Q…           5 PhosphoSite:16319926;ProtMapper:1…
## 4 PRKG1 (Q… ==( - )==&gt;  TRPC3 (Q…           8 HPRD:14983059;KEA:14983059;ProtMa…
## 5 TRPM7 (Q… ==( + )==&gt;  ANXA1 (P…          10 HPRD-phos:15485879;HPRD-phos:2006…
## 6 PTPN1 (P… ==( - )==&gt;  TRPV6 (Q…           6 DEPOD:15894168;DEPOD:17197020;HPR…</code></pre>
<div id="protein-protein-interaction-networks" class="section level3">
<h3 class="hasAnchor">
<a href="#protein-protein-interaction-networks" class="anchor" aria-hidden="true"></a>Protein-protein interaction networks</h3>
<p>Protein-protein interactions are usually converted into networks. Describing protein interactions as networks not only provides a convenient format for visualization, but also allows applying graph theory methods to mine the biological information they contain.</p>
<p>We convert here our set of interactions to a network/graph (<em><a href="https://CRAN.R-project.org/package=igraph" class="external-link">igraph</a></em>object). Then, we apply two very common approaches to extract information from a biological network:</p>
<ul>
<li>
<strong>Shortest Paths</strong>: finding a path between two nodes (proteins) going through the minimum number of edges. This can be very useful to track consecutive reactions within a given pathway. We display below the shortest path between two given proteins and all the possible shortests paths between two other proteins. It is to note that the functions <code>print_path\_es</code> and <code>print_path\_vs</code> display very similar results, but the first one takes as an input an edge sequence and the second one a node sequence.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We transform the interactions data frame into a graph</span>
<span class="va">OPI_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span>interactions <span class="op">=</span> <span class="va">interactions</span><span class="op">)</span>

<span class="co">## Find and print shortest paths on the directed network between proteins</span>
<span class="co">## of interest:</span>
<span class="fu"><a href="../reference/print_path_es.html">print_path_es</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/distances.html" class="external-link">shortest_paths</a></span><span class="op">(</span><span class="va">OPI_g</span>,from <span class="op">=</span> <span class="st">"TYRO3"</span>,to <span class="op">=</span> <span class="st">"STAT3"</span>,
    output <span class="op">=</span> <span class="st">'epath'</span><span class="op">)</span><span class="op">$</span><span class="va">epath</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,<span class="va">OPI_g</span><span class="op">)</span></code></pre></div>
<pre><code>##           source interaction         target n_resources
## 1 TYRO3 (Q06418)  ==( + )==&gt;  GRB2 (P62993)           2
## 2  GRB2 (P62993)  ==( + )==&gt;  EGFR (P00533)          11
## 3  EGFR (P00533)  ==( + )==&gt; STAT3 (P40763)          12
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           n_references
## 1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      SIGNOR:17306385
## 2 DIP:1322798;DOMINO:12577067;DOMINO:16273093;DOMINO:16729043;HPRD:11726515;HPRD:12577067;HPRD:1322798;HPRD:14679214;HPRD:16799092;HPRD:7518560;HPRD:7527043;HPRD:8662998;HPRD:8810325;HPRD:8887653;InnateDB:16799092;IntAct:10229072;IntAct:11896612;IntAct:14679214;IntAct:16273093;IntAct:16729043;IntAct:21278786;IntAct:23799367;IntAct:24658140;IntAct:27059931;IntAct:7518560;IntAct:7527043;IntAct:8887653;IntAct:8940013;IntAct:9020117;LMPID:7518560;LMPID:8491186;Lit-BM-17:10026169;Lit-BM-17:10085134;Lit-BM-17:10229072;Lit-BM-17:10595738;Lit-BM-17:11370743;Lit-BM-17:11408594;Lit-BM-17:11896612;Lit-BM-17:11960376;Lit-BM-17:12577067;Lit-BM-17:1322798;Lit-BM-17:14679214;Lit-BM-17:15657067;Lit-BM-17:16273093;Lit-BM-17:16729043;Lit-BM-17:16799092;Lit-BM-17:17242169;Lit-BM-17:17971399;Lit-BM-17:18271526;Lit-BM-17:19289468;Lit-BM-17:19531499;Lit-BM-17:19836242;Lit-BM-17:20124286;Lit-BM-17:21596750;Lit-BM-17:21706016;Lit-BM-17:23178716;Lit-BM-17:23799367;Lit-BM-17:23956138;Lit-BM-17:24189400;Lit-BM-17:24658140;Lit-BM-17:24797263;Lit-BM-17:25402006;Lit-BM-17:27059931;Lit-BM-17:7506413;Lit-BM-17:7510700;Lit-BM-17:7518560;Lit-BM-17:7527043;Lit-BM-17:7592693;Lit-BM-17:7650032;Lit-BM-17:8647858;Lit-BM-17:8662998;Lit-BM-17:8810340;Lit-BM-17:8887653;Lit-BM-17:8940013;Lit-BM-17:9020117;Lit-BM-17:9050991;Lit-BM-17:9416834;Lit-BM-17:9886492;NetPath:12577067;NetPath:17991782;NetPath:19509291;NetPath:19531499;NetPath:9544989;SIGNOR:11726515;SPIKE:11084343;SPIKE:12648464;SPIKE:16713569;SPIKE:7518560;SPIKE:8887653
## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   BioGRID:18258752;BioGRID:20145033;CancerCellMap:12070153;HPRD:12873986;InnateDB:12873986;InnateDB:14963038;InnateDB:15284024;InnateDB:15950906;IntAct:14966128;IntAct:24135280;IntAct:24658140;IntAct:25241761;Lit-BM-17:10358079;Lit-BM-17:14966128;Lit-BM-17:15284024;Lit-BM-17:15485908;Lit-BM-17:15657067;Lit-BM-17:15950906;Lit-BM-17:18258752;Lit-BM-17:21573184;Lit-BM-17:23956138;Lit-BM-17:24135280;Lit-BM-17:24189400;Lit-BM-17:24658140;Lit-BM-17:25241761;Lit-BM-17:25402006;Lit-BM-17:9756944;NetPath:20032199;ProtMapper:15951569;ProtMapper:23894143;SIGNOR:14967450;SIGNOR:22693070;SPIKE:15950906</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Find and print all shortest paths between proteins of interest:</span>
<span class="fu"><a href="../reference/print_path_vs.html">print_path_vs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/distances.html" class="external-link">all_shortest_paths</a></span><span class="op">(</span><span class="va">OPI_g</span>,from <span class="op">=</span> <span class="st">"DYRK2"</span>,
    to <span class="op">=</span> <span class="st">"MAPKAPK2"</span><span class="op">)</span><span class="op">$</span><span class="va">res</span>,<span class="va">OPI_g</span><span class="op">)</span></code></pre></div>
<pre><code>## Pathway 1: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PEA15 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 2: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PEA15 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 3: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PEA15 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 4: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PPP1CA -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 5: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PPP1CA -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 6: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PPP1CA -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 7: DYRK2 -&gt; TP53 -&gt; RPS6KA5 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 8: DYRK2 -&gt; TP53 -&gt; RPS6KA4 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 9: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 10: DYRK2 -&gt; TP53 -&gt; RPS6KA2 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 11: DYRK2 -&gt; TP53 -&gt; RPS6KA1 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 12: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PEA15 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 13: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PEA15 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 14: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PEA15 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 15: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PPP1CA -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 16: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PPP1CA -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 17: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PPP1CA -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 18: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; DAPK1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 19: DYRK2 -&gt; TP53 -&gt; RPS6KA5 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 20: DYRK2 -&gt; TP53 -&gt; RPS6KA4 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 21: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 22: DYRK2 -&gt; TP53 -&gt; RPS6KA2 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 23: DYRK2 -&gt; TP53 -&gt; RPS6KA1 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 24: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 25: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 26: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 27: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP2K4 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 28: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP2K4 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 29: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP2K4 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 30: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 31: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 32: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<ul>
<li>
<strong>Clustering</strong>: grouping nodes (proteins) in such a way that nodes belonging to the same group (called cluster) are more connected in the network to each other than to those in other groups (clusters). Since proteins interact to perform their functions, proteins within the same cluster are likely to be implicated in similar biological tasks. Figure @ref(fig:fig2) shows the subgraph containing the proteins and interactions of a specifc protein, ERBB2 The <em><a href="https://CRAN.R-project.org/package=igraph" class="external-link">igraph</a></em> package contains functions to apply sevaral different cluster methods on graphs (visit <a href="https://igraph.org/r/doc/" class="external-link uri">https://igraph.org/r/doc/</a> for detailed information.)</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We apply a clustering algorithm (Louvain) to group proteins in</span>
<span class="co">## our network. We apply here Louvain which is fast but can only run</span>
<span class="co">## on undirected graphs. Other clustering algorithms can deal with</span>
<span class="co">## directed networks but with longer computational times,</span>
<span class="co">## such as cluster_edge_betweenness. These cluster methods are directly</span>
<span class="co">## available in the igraph package.</span>
<span class="va">OPI_g_undirected</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/as.directed.html" class="external-link">as.undirected</a></span><span class="op">(</span><span class="va">OPI_g</span>, mode<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mutual"</span><span class="op">)</span><span class="op">)</span>
<span class="va">OPI_g_undirected</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/simplify.html" class="external-link">simplify</a></span><span class="op">(</span><span class="va">OPI_g_undirected</span><span class="op">)</span>
<span class="va">cl_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_fast_greedy.html" class="external-link">cluster_fast_greedy</a></span><span class="op">(</span><span class="va">OPI_g_undirected</span><span class="op">)</span>
<span class="co">## We extract the cluster where a protein of interest is contained</span>
<span class="va">cluster_id</span> <span class="op">&lt;-</span> <span class="va">cl_results</span><span class="op">$</span><span class="va">membership</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">cl_results</span><span class="op">$</span><span class="va">names</span> <span class="op">==</span> <span class="st">"ERBB2"</span><span class="op">)</span><span class="op">]</span>
<span class="va">module_graph</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/subgraph.html" class="external-link">induced_subgraph</a></span><span class="op">(</span><span class="va">OPI_g_undirected</span>,
    <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">OPI_g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">cl_results</span><span class="op">$</span><span class="va">membership</span> <span class="op">==</span> <span class="va">cluster_id</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><code>{r fig2, fig.width=10, fig.height=5, dpi=300, echo = FALSE,     fig.cap="ERBB2 associated cluser. Subnetwork extracted from the     interactions graph representing the cluster where we can find the     gene *ERBB2* (yellow node)"} ## We print that cluster with its interactions. par(mar=c(0.1,0.1,0.1,0.1)) plot(module_graph, vertex.label.color="black",vertex.frame.color="#ffffff",     vertex.size= 15, edge.curved=.2,     vertex.color = ifelse(igraph::V(module_graph)$name == "ERBB2","yellow",     "#00CCFF"), edge.color="blue",edge.width=0.8)</code></p>
</div>
</div>
<div id="other-interaction-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#other-interaction-datasets" class="anchor" aria-hidden="true"></a>Other interaction datasets</h2>
<p>We used above the interactions from the dataset described in the original <strong>OmniPath</strong> publication <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>. In this section, we provide examples on how to retry and deal with interactions from the remaining datasets. The same functions can been applied to every interaction dataset.</p>
<div id="pathway-extra" class="section level3">
<h3 class="hasAnchor">
<a href="#pathway-extra" class="anchor" aria-hidden="true"></a>Pathway Extra</h3>
<p>In the first example, we are going to get the interactions from the <strong>pathwayextra</strong> dataset, which contains activity flow interactions without literature reference. We are going to focus on the mouse interactions for a given gene in this particular case.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="va">interactions</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/import_pathwayextra_interactions.html">import_pathwayextra_interactions</a></span><span class="op">(</span>resources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BioGRID"</span>,<span class="st">"STRING"</span><span class="op">)</span>,
    organism <span class="op">=</span> <span class="fl">10090</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in omnipath_check_param(param): The following resources are not
## available: STRING. Check the resource names for spelling mistakes.</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We select all the interactions in which Amfr gene is involved</span>
<span class="va">interactions_Amfr</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">interactions</span>, <span class="va">source_genesymbol</span> <span class="op">==</span> <span class="st">"Amfr"</span> <span class="op">|</span>
    <span class="va">target_genesymbol</span> <span class="op">==</span> <span class="st">"Amfr"</span><span class="op">)</span>

<span class="co">## We print these interactions:</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">interactions_Amfr</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 1 x 5
##   source    interaction target   n_resources n_references                       
##   &lt;chr&gt;     &lt;chr&gt;       &lt;chr&gt;          &lt;int&gt; &lt;chr&gt;                              
## 1 Amfr (Q9… ==( + )==&gt;  Vcp (Q0…           6 BioGRID:16987818;BioGRID:17311810;…</code></pre>
</div>
<div id="kinase-extra" class="section level3">
<h3 class="hasAnchor">
<a href="#kinase-extra" class="anchor" aria-hidden="true"></a>Kinase Extra</h3>
<p>Next, we download the interactions from the <strong>kinaseextra</strong> dataset, which contains enzyme-substrate interactions without literature reference. We are going to focus on rat reactions targeting a particular gene.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="va">interactions</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/import_kinaseextra_interactions.html">import_kinaseextra_interactions</a></span><span class="op">(</span>resources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PhosphoPoint"</span>,
    <span class="st">"PhosphoSite"</span><span class="op">)</span>, organism <span class="op">=</span> <span class="fl">10116</span><span class="op">)</span>

<span class="co">## We select the interactions in which Dpysl2 gene is a target</span>
<span class="va">interactions_TargetDpysl2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">interactions</span>,
    <span class="va">target_genesymbol</span> <span class="op">==</span> <span class="st">"Dpysl2"</span><span class="op">)</span>

<span class="co">## We print these interactions:</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">interactions_TargetDpysl2</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 5 x 5
##   source    interaction target    n_resources n_references                      
##   &lt;chr&gt;     &lt;chr&gt;       &lt;chr&gt;           &lt;int&gt; &lt;chr&gt;                             
## 1 Fer (P09… ==( ? )==&gt;  Dpysl2 (…           2 PhosphoSite:29396402;ProtMapper:2…
## 2 Cdk5 (Q0… ==( + )==&gt;  Dpysl2 (…           5 PhosphoSite:15652488;PhosphoSite:…
## 3 Rock1 (Q… ==( ? )==&gt;  Dpysl2 (…           6 HPRD:10818093;HPRD:12482610;KEA:1…
## 4 Gsk3b (P… ==(+/-)==&gt;  Dpysl2 (…          11 HPRD-phos:15652488;HPRD-phos:1696…
## 5 Rock2 (Q… ==( + )==&gt;  Dpysl2 (…          10 CA1:10818093;KEA:10818093;Phospho…</code></pre>
</div>
<div id="ligand-receptor-extra" class="section level3">
<h3 class="hasAnchor">
<a href="#ligand-receptor-extra" class="anchor" aria-hidden="true"></a>Ligand-receptor Extra</h3>
<p>In the following example we are going to work with the <strong>ligrecextra</strong> dataset, which contains ligand-receptor interactions without literature reference. Our goal is to find the potential receptors associated to a given ligand, CDH1 (Figure @ref(fig:fig3)).</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="va">interactions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_ligrecextra_interactions.html">import_ligrecextra_interactions</a></span><span class="op">(</span>resources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"iTALK"</span>,
    <span class="st">"Baccin2019"</span><span class="op">)</span>, organism<span class="op">=</span><span class="fl">9606</span><span class="op">)</span>

<span class="co">## Receptors of the CDH1 ligand.</span>
<span class="va">interactions_ADM2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">interactions</span>, <span class="va">source_genesymbol</span> <span class="op">==</span> <span class="st">"ADM2"</span><span class="op">)</span>

<span class="co">## We transform the interactions data frame into a graph</span>
<span class="va">OPI_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span>interactions <span class="op">=</span> <span class="va">interactions_ADM2</span><span class="op">)</span>

<span class="co">## We induce a network with these genes</span>
<span class="va">Induced_Network</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/pkg/dnet/man/dNetInduce.html" class="external-link">dNetInduce</a></span><span class="op">(</span>g<span class="op">=</span><span class="va">OPI_g</span>,
    nodes_query<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">OPI_g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>, knn<span class="op">=</span><span class="fl">0</span>,
    remove.loops<span class="op">=</span><span class="cn">FALSE</span>, largest.comp<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><code>{r fig3, dpi=300, echo = FALSE, fig.cap="Ligand-receptor interactions for     the ADM2 ligand."} ## We print the induced network par(mar=c(0.1,0.1,0.1,0.1)) plot(Induced_Network, vertex.label.color="black",     vertex.frame.color="#ffffff",vertex.size= 20, edge.curved=.2,     vertex.color =         ifelse(igraph::V(Induced_Network)$name %in% c("ADM2"),         "yellow","#00CCFF"), edge.color="blue",edge.width=0.8)</code></p>
</div>
<div id="dorothea-regulons" class="section level3">
<h3 class="hasAnchor">
<a href="#dorothea-regulons" class="anchor" aria-hidden="true"></a>DoRothEA Regulons</h3>
<p>Another very interesting interaction dataset also available in OmniPath is <strong>DoRothEA</strong> <span class="citation">(Garcia-Alonso et al. 2019)</span>. It contains transcription factor (TF)-target interactions with confidence score, ranging from A-E, being A the most confident interactions. In the code chunk shown below, we select and print the most confident interactions for a given TF.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="va">interactions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_dorothea_interactions.html">import_dorothea_interactions</a></span><span class="op">(</span>
    resources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DoRothEA"</span><span class="op">)</span>,
    dorothea_levels <span class="op">=</span> <span class="st">'A'</span>,
    organism<span class="op">=</span><span class="fl">9606</span>
<span class="op">)</span>

<span class="co">## We select the most confident interactions for a given TF and we print</span>
<span class="co">## the interactions to check the way it regulates its different targets</span>
<span class="va">interactions_A_GLI1</span>  <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">interactions</span>, <span class="va">dorothea_level</span><span class="op">==</span><span class="st">"A"</span>,
    <span class="va">source_genesymbol</span> <span class="op">==</span> <span class="st">"GLI1"</span><span class="op">)</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">interactions_A_GLI1</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 7 x 5
##   source        interaction target          n_resources n_references
##   &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;                 &lt;int&gt; &lt;chr&gt;       
## 1 GLI1 (P08151) ==( + )==&gt;  PTCH1 (Q13635)            1 0           
## 2 GLI1 (P08151) ==( + )==&gt;  BCL2 (P10415)             0 0           
## 3 GLI1 (P08151) ==( + )==&gt;  CCND2 (P30279)            0 0           
## 4 GLI1 (P08151) ==( - )==&gt;  EGR2 (P11161)             0 0           
## 5 GLI1 (P08151) ==( + )==&gt;  IGFBP6 (P24592)           0 0           
## 6 GLI1 (P08151) ==( + )==&gt;  SFRP1 (Q8N474)            0 0           
## 7 GLI1 (P08151) ==( - )==&gt;  SLIT2 (O94813)            0 0</code></pre>
</div>
<div id="mirna-target-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#mirna-target-dataset" class="anchor" aria-hidden="true"></a>miRNA-target dataset</h3>
<p>The last dataset describing interactions is <strong>mirnatarget</strong>. It stores miRNA-mRNA and TF-miRNA interactions. These interactions are only available for human so far. We next select the miRNA interacting with the TF selected in the previous code chunk, <em>GLI1</em>. The main function of miRNAs seems to be related with gene regulation. It is therefore interesting to see how some miRNA can regulate the expression of a TF which in turn regulates the expression of other genes. Figure @ref(fig:fig4) shows a schematic network of the miRNA targeting <em>GLI1</em> and the genes regulated by this TF.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="va">interactions</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/import_mirnatarget_interactions.html">import_mirnatarget_interactions</a></span><span class="op">(</span>resources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"miRTarBase"</span>,<span class="st">"miRecords"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## We select the interactions where a miRNA is interacting with the TF</span>
<span class="co">## used in the previous code chunk and we print these interactions.</span>
<span class="va">interactions_miRNA_GLI1</span> <span class="op">&lt;-</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">interactions</span>,  <span class="va">target_genesymbol</span> <span class="op">==</span> <span class="st">"GLI1"</span><span class="op">)</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">interactions_miRNA_GLI1</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 5 x 5
##   source          interaction target    n_resources n_references                
##   &lt;chr&gt;           &lt;chr&gt;       &lt;chr&gt;           &lt;int&gt; &lt;chr&gt;                       
## 1 hsa-miR-133b (… ==( ? )==&gt;  GLI1 (P0…           1 miRTarBase:24443799         
## 2 hsa-miR-202 (M… ==( ? )==&gt;  GLI1 (P0…           1 miRTarBase:23936094         
## 3 hsa-miR-324-5p… ==( ? )==&gt;  GLI1 (P0…           3 miRTarBase:18756266;miRTarB…
## 4 hsa-miR-125b (… ==( ? )==&gt;  GLI1 (P0…           2 miRTarBase:18756266         
## 5 hsa-miR-326 (M… ==( ? )==&gt;  GLI1 (P0…           2 miRTarBase:18756266</code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We transform the previous selections to graphs (igraph objects)</span>
<span class="va">OPI_g_1</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span>interactions <span class="op">=</span> <span class="va">interactions_A_GLI1</span><span class="op">)</span>
<span class="va">OPI_g_2</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span>interactions <span class="op">=</span> <span class="va">interactions_miRNA_GLI1</span><span class="op">)</span></code></pre></div>
<p><code>{r fig4, dpi=300, echo = FALSE, fig.cap="miRNA-TF-target network. Schematic network of the miRNA (red square nodes) targeting \textit{GLI1} (yellow node) and the genes regulated by this TF (blue round nodes)."} ## We print the union of both previous graphs par(mar=c(0.1,0.1,0.1,0.1)) plot(OPI_g_1 %u% OPI_g_2, vertex.label.color="black",     vertex.frame.color="#ffffff",vertex.size= 20, edge.curved=.25,     vertex.color = ifelse(grepl("miR",igraph::V(OPI_g_1 %u% OPI_g_2)$name),     "red",ifelse(igraph::V(OPI_g_1 %u% OPI_g_2)$name == "GLI1",     "yellow","#00CCFF")), edge.color="blue",     vertex.shape = ifelse(grepl("miR",igraph::V(OPI_g_1 %u% OPI_g_2)$name),     "vrectangle","circle"),edge.width=0.8)</code></p>
</div>
</div>
<div id="post-translational-modifications-ptms" class="section level2">
<h2 class="hasAnchor">
<a href="#post-translational-modifications-ptms" class="anchor" aria-hidden="true"></a>Post-translational modifications (PTMs)</h2>
<p>Another query type available is PTMs which provides enzyme-substrate reactions in a very similar way to the aforementioned interactions. PTMs refer generally to enzymatic modification of proteins after their synthesis in the ribosomes. PTMs can be highly context-specific and they play a main role in the activation/inhibition of biological pathways.</p>
<p>In the next code chunk, we download the <strong>PTMs</strong> for human. We first check the different available source databases, even though we do not perform any filter. Then, we select and print the reactions involving a specific enzyme-substrate pair. Those reactions lack information about activation or inhibition. To obtain that information, we match the data with <strong>OmniPath</strong> interactions. Finally, we show that it is also possible to build a graph using this information, and to retrieve PTMs from mouse or rat.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We check the different PTMs databases</span>
<span class="fu"><a href="../reference/get_enzsub_resources.html">get_enzsub_resources</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "BEL-Large-Corpus_ProtMapper" "dbPTM"                      
##  [3] "DEPOD"                       "HPRD"                       
##  [5] "HPRD_MIMP"                   "KEA"                        
##  [7] "Li2012"                      "MIMP"                       
##  [9] "NCI-PID_ProtMapper"          "phosphoELM"                 
## [11] "phosphoELM_MIMP"             "PhosphoNetworks"            
## [13] "PhosphoSite"                 "PhosphoSite_MIMP"           
## [15] "PhosphoSite_ProtMapper"      "ProtMapper"                 
## [17] "REACH_ProtMapper"            "Reactome_ProtMapper"        
## [19] "RLIMS-P_ProtMapper"          "SIGNOR"                     
## [21] "SIGNOR_ProtMapper"           "Sparser_ProtMapper"</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We query and store the enzyme-PTM interactions into a dataframe.</span>
<span class="co">## No filtering by databases in this case.</span>
<span class="va">enzsub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_enzsub.html">import_omnipath_enzsub</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">## We can select and print the reactions between a specific kinase and</span>
<span class="co">## a specific substrate</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span>
    <span class="va">enzsub</span>,
    <span class="va">enzyme_genesymbol</span> <span class="op">==</span> <span class="st">"MAP2K1"</span>,
    <span class="va">substrate_genesymbol</span> <span class="op">==</span> <span class="st">"MAPK3"</span>
<span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Unknown or uninitialised column: `is_stimulation`.</code></pre>
<pre><code>## # A tibble: 6 x 5
##   enzyme          interaction substrate           modification    n_resources
##   &lt;chr&gt;           &lt;chr&gt;       &lt;chr&gt;               &lt;chr&gt;                 &lt;int&gt;
## 1 MAP2K1 (Q02750) ====&gt;       MAPK3_T202 (P27361) phosphorylation           8
## 2 MAP2K1 (Q02750) ====&gt;       MAPK3_Y204 (P27361) phosphorylation           8
## 3 MAP2K1 (Q02750) ====&gt;       MAPK3_Y210 (P27361) phosphorylation           2
## 4 MAP2K1 (Q02750) ====&gt;       MAPK3_T207 (P27361) phosphorylation           2
## 5 MAP2K1 (Q02750) ====&gt;       MAPK3_T80 (P27361)  phosphorylation           1
## 6 MAP2K1 (Q02750) ====&gt;       MAPK3_Y222 (P27361) phosphorylation           1</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## In the previous results, we can see that enzyme-PTM relationships do not</span>
<span class="co">## contain sign (activation/inhibition). We can generate this information</span>
<span class="co">## based on the protein-protein OmniPath interaction dataset.</span>
<span class="va">interactions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">enzsub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_signed_ptms.html">get_signed_ptms</a></span><span class="op">(</span><span class="va">enzsub</span>, <span class="va">interactions</span><span class="op">)</span>

<span class="co">## We select again the same kinase and substrate. Now we have information</span>
<span class="co">## about inhibition or activation when we print the enzyme-PTM relationships</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">enzsub</span>,<span class="va">enzyme_genesymbol</span><span class="op">==</span><span class="st">"MAP2K1"</span>,
    <span class="va">substrate_genesymbol</span><span class="op">==</span><span class="st">"MAPK3"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##            enzyme interaction           substrate    modification n_resources
## 5 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_T202 (P27361) phosphorylation           8
## 1 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y204 (P27361) phosphorylation           8
## 2 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_T207 (P27361) phosphorylation           2
## 3 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y210 (P27361) phosphorylation           2
## 4 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y222 (P27361) phosphorylation           1
## 6 MAP2K1 (Q02750)  ==( + )==&gt;  MAPK3_T80 (P27361) phosphorylation           1</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We can also transform the enzyme-PTM relationships into a graph.</span>
<span class="va">enzsub_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enzsub_graph.html">enzsub_graph</a></span><span class="op">(</span>enzsub <span class="op">=</span> <span class="va">enzsub</span><span class="op">)</span>

<span class="co">## We download PTMs for mouse</span>
<span class="va">enzsub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_enzsub.html">import_omnipath_enzsub</a></span><span class="op">(</span>
    resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PhosphoSite"</span>, <span class="st">"SIGNOR"</span><span class="op">)</span>,
    organism <span class="op">=</span> <span class="fl">10090</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="complexes" class="section level2">
<h2 class="hasAnchor">
<a href="#complexes" class="anchor" aria-hidden="true"></a>Complexes</h2>
<p>Some studies indicate that around 80% of the human proteins operate in complexes, and many proteins belong to several different complexes <span class="citation">(Berggård, Linse, and James 2007)</span>. These complexes play critical roles in a large variety of biological processes. Some well-known examples are the proteasome and the ribosome. Thus, the description of the full set of protein complexes functioning in cells is essential to improve our understanding of biological processes.</p>
<p>The <strong>complexes</strong> query provides access to more than 20000 protein complexes. This comprehensive database has been created by integrating different resources. We now download these molecular complexes filtering by some of the source databases. We check the complexes where a couple of specific genes participate. First, we look for the complexes where any of these two genes participate. We then identify the complex where these two genes are jointly involved. Finally, we perform an enrichment analysis with the genes taking part in that complex. You should keep an eye on this complex since it will be used again in the forthcoming sections.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We check the different complexes databases</span>
<span class="fu"><a href="../reference/get_complex_resources.html">get_complex_resources</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "CellChatDB"     "CellPhoneDB"    "CFinder"        "Compleat"      
##  [5] "ComplexPortal"  "CORUM"          "Guide2Pharma"   "Havugimana2012"
##  [9] "HPMR"           "hu.MAP"         "hu.MAP2"        "ICELLNET"      
## [13] "KEGG-MEDICUS"   "NetworkBlast"   "PDB"            "Signor"        
## [17] "SIGNOR"</code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We query and store complexes from some sources into a dataframe.</span>
<span class="va">complexes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_complexes.html">import_omnipath_complexes</a></span><span class="op">(</span>resources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CORUM"</span>, <span class="st">"hu.MAP"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## We check all the molecular complexes where a set of genes participate</span>
<span class="va">query_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WRN"</span>,<span class="st">"PARP1"</span><span class="op">)</span>

<span class="co">## Complexes where any of the input genes participate</span>
<span class="va">complexes_query_genes_any</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_complex_genes.html">get_complex_genes</a></span><span class="op">(</span><span class="va">complexes</span>,<span class="va">query_genes</span>,
    total_match<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>

<span class="co">## We print the components of the different selected components</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">complexes_query_genes_any</span><span class="op">$</span><span class="va">components_genesymbols</span>,<span class="fl">6</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "NCAPD2_NCAPG_NCAPH_PARP1_SMC2_SMC4_XRCC1"                             
## [2] "CCNA2_CDK2_LIG1_PARP1_POLA1_POLD1_POLE_RFC1_RFC2_RPA1_RPA2_RPA3_TOP1" 
## [3] "CCNA2_CCNB1_CDK1_PARP1_POLA1_POLD1_POLE_RFC1_RFC2_RPA1_RPA2_RPA3_TOP1"
## [4] "MRE11_PARP1_RAD50_TERF2_TERF2IP_XRCC5_XRCC6"                          
## [5] "TERF2_WRN"                                                            
## [6] "CALR_DHX30_H2AFX_HIST3H2BB_HSPA5_NPM1_PARP1"</code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Complexes where all the input genes participate jointly</span>
<span class="va">complexes_query_genes_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_complex_genes.html">get_complex_genes</a></span><span class="op">(</span><span class="va">complexes</span>,<span class="va">query_genes</span>,
    total_match<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>

<span class="co">## We print the components of the different selected components</span>
<span class="va">complexes_query_genes_join</span><span class="op">$</span><span class="va">components_genesymbols</span></code></pre></div>
<pre><code>## [1] "PARP1_WRN_XRCC5_XRCC6"</code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes_complex</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">complexes_query_genes_join</span><span class="op">$</span><span class="va">components_genesymbols</span>, <span class="st">"_"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## We can perform an enrichment analyses with the genes in the complex</span>
<span class="va">EnrichmentResults</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gost</a></span><span class="op">(</span><span class="va">genes_complex</span>, significant <span class="op">=</span> <span class="cn">TRUE</span>,
    user_threshold <span class="op">=</span> <span class="fl">0.001</span>, correction_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fdr"</span><span class="op">)</span>,
    sources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:BP"</span>,<span class="st">"GO:CC"</span>,<span class="st">"GO:MF"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## We show the most significant results</span>
<span class="va">EnrichmentResults</span><span class="op">$</span><span class="va">result</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">term_id</span>, <span class="va">source</span>, <span class="va">term_name</span>,<span class="va">p_value</span><span class="op">)</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">5</span>,<span class="op">-</span><span class="va">p_value</span><span class="op">)</span></code></pre></div>
<pre><code>##      term_id source                      term_name      p_value
## 1 GO:0010332  GO:BP    response to gamma radiation 4.819842e-08
## 2 GO:0032508  GO:BP           DNA duplex unwinding 3.143404e-07
## 3 GO:0032392  GO:BP           DNA geometric change 3.143404e-07
## 4 GO:0010212  GO:BP response to ionizing radiation 6.780115e-07
## 5 GO:0000781  GO:CC   chromosome, telomeric region 3.221331e-07</code></pre>
</div>
<div id="annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#annotations" class="anchor" aria-hidden="true"></a>Annotations</h2>
<p>Biological annotations are statements, usually traceable and curated, about the different features of a biological entity. At the genetic level, annotations describe the biological function, the subcellular situation, the DNA location and many other related properties of a particular gene or its gene products.</p>
<p>The annotations query provides a large variety of data about proteins and complexes. These data come from dozens of databases and each kind of annotation record contains different fields. Because of this, here we have a record_id field which is unique within the records of each database. Each row contains one key value pair and you need to use the record_id to connect the related key-value pairs (see examples below).</p>
<p>Now, we focus in the annotations of the complex studied in the previous section. We first inspect the different available databases in the omnipath webserver. Then, we download the annotations for our complex itself as a biological entity. We find annotations related to the nucleus and transcriptional control, which is in agreement with the enrichment analysis results of its individual components.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We check the different annotation databases</span>
<span class="fu"><a href="../reference/get_annotation_resources.html">get_annotation_resources</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "Adhesome"             "Almen2009"            "Baccin2019"          
##  [4] "CancerGeneCensus"     "CancerSEA"            "CellCellInteractions"
##  [7] "CellChatDB"           "CellChatDB_complex"   "CellPhoneDB"         
## [10] "CellPhoneDB_complex"  "CellTalkDB"           "ComPPI"              
## [13] "connectomeDB2020"     "CORUM_Funcat"         "CORUM_GO"            
## [16] "CSPA"                 "CSPA_celltype"        "DGIdb"               
## [19] "DisGeNet"             "EMBRACE"              "Exocarta"            
## [22] "GO_Intercell"         "GPCRdb"               "Guide2Pharma"        
## [25] "HGNC"                 "HPA_secretome"        "HPA_subcellular"     
## [28] "HPA_tissue"           "HPMR"                 "HPMR_complex"        
## [31] "HumanCellMap"         "ICELLNET"             "ICELLNET_complex"    
## [34] "Integrins"            "IntOGen"              "iTALK"               
## [37] "KEGG-PC"              "kinase.com"           "Kirouac2010"         
## [40] "LOCATE"               "LRdb"                 "Matrisome"           
## [43] "MatrixDB"             "MCAM"                 "Membranome"          
## [46] "MSigDB"               "NetPath"              "OPM"                 
## [49] "Phobius"              "Phosphatome"          "Ramilowski_location" 
## [52] "Ramilowski2015"       "SignaLink_function"   "SignaLink_pathway"   
## [55] "SIGNOR"               "Surfaceome"           "talklr"              
## [58] "TCDB"                 "TFcensus"             "TopDB"               
## [61] "UniProt_family"       "UniProt_keyword"      "UniProt_location"    
## [64] "UniProt_tissue"       "UniProt_topology"     "Vesiclepedia"        
## [67] "Zhong2015"</code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We can further investigate the features of the complex selected</span>
<span class="co">## in the previous section.</span>

<span class="co">## We first get the annotations of the complex itself:</span>
<span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span>proteins<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"COMPLEX:"</span>,
  <span class="va">complexes_query_genes_join</span><span class="op">$</span><span class="va">components_genesymbols</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">annotations</span>,<span class="va">source</span>,<span class="va">label</span>,<span class="va">value</span><span class="op">)</span>,<span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 10 x 3
##    source           label      value                             
##    &lt;chr&gt;            &lt;chr&gt;      &lt;chr&gt;                             
##  1 LOCATE           location   nucleus                           
##  2 LOCATE           location   nucleolus                         
##  3 UniProt_location location   Nucleus                           
##  4 CancerSEA        state      DNA repair                        
##  5 MSigDB           collection reactome_pathways                 
##  6 MSigDB           geneset    REACTOME_DNA_REPAIR               
##  7 MSigDB           collection chemical_and_genetic_perturbations
##  8 MSigDB           geneset    PUJANA_CHEK2_PCC_NETWORK          
##  9 MSigDB           collection chemical_and_genetic_perturbations
## 10 MSigDB           geneset    KAUFFMANN_DNA_REPAIR_GENES</code></pre>
<p>Afterwards, we explore the annotations of the individual components of the complex in some databases. We check the pathways where these proteins are involved. Once again, we also find many nucleus related annotations when checking their cellular location.</p>
<p>Then, we explore some annotations of its individual components. Pathways where the proteins belong:</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span>
    proteins <span class="op">=</span> <span class="va">genes_complex</span>,
    resources <span class="op">=</span> <span class="st">"NetPath"</span>
<span class="op">)</span>

<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="va">genesymbol</span>, <span class="va">value</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 7 x 2
##   genesymbol value                                        
##   &lt;chr&gt;      &lt;chr&gt;                                        
## 1 PARP1      Corticotropin-releasing hormone (CRH)        
## 2 PARP1      TNF-related weak inducer of apoptosis (TWEAK)
## 3 PARP1      Oncostatin-M (OSM)                           
## 4 PARP1      Androgen receptor (AR)                       
## 5 PARP1      Tumor necrosis factor (TNF) alpha            
## 6 XRCC5      Androgen receptor (AR)                       
## 7 XRCC6      Androgen receptor (AR)</code></pre>
<p>Subcellular localization of our proteins:</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotations</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span>
    proteins <span class="op">=</span> <span class="va">genes_complex</span>,
    resources <span class="op">=</span> <span class="st">"ComPPI"</span>
<span class="op">)</span></code></pre></div>
<p>Since we have same record_id for some results of our query, we spread these records across columns:</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/spread.html" class="external-link">spread</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="va">label</span>, <span class="va">value</span><span class="op">)</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">10</span>, <span class="va">score</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 11 x 7
##    uniprot genesymbol entity_type source record_id location      score          
##    &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;         &lt;chr&gt;          
##  1 P12956  XRCC6      protein     ComPPI      2857 nucleus       0.999999976291…
##  2 P09874  PARP1      protein     ComPPI     11110 nucleus       0.999999887104 
##  3 Q14191  WRN        protein     ComPPI     16012 nucleus       0.9999996544   
##  4 P13010  XRCC5      protein     ComPPI     13276 nucleus       0.99999868288  
##  5 P13010  XRCC5      protein     ComPPI     13275 membrane      0.972          
##  6 P12956  XRCC6      protein     ComPPI      2860 cytosol       0.958          
##  7 P13010  XRCC5      protein     ComPPI     13277 cytosol       0.958          
##  8 Q14191  WRN        protein     ComPPI     16011 cytosol       0.94           
##  9 P12956  XRCC6      protein     ComPPI      2858 membrane      0.860000000000…
## 10 P12956  XRCC6      protein     ComPPI      2859 extracellular 0.860000000000…
## 11 P13010  XRCC5      protein     ComPPI     13274 extracellular 0.860000000000…</code></pre>
<p>The way above, we more or less reconstituted the data as it is in the original resource. The same can be done much easier by passing the <code>wide = TRUE</code> parameter to <code>import_omnipath_annotations</code>. In this case, if the data contains more than one resources, a list of data frames will be returned.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">signaling_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span>
    resources <span class="op">=</span> <span class="st">'SignaLink_pathway'</span>,
    wide <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="intercell" class="section level2">
<h2 class="hasAnchor">
<a href="#intercell" class="anchor" aria-hidden="true"></a>Intercell</h2>
<p>Cells perceive cues from their microenvironment and neighboring cells, and respond accordingly to ensure proper activities and coordination between them. The ensemble of these communication process is called inter-cellular signaling (<strong>intercell</strong>).</p>
<p><strong>Intercell</strong> query provides information about the roles of proteins in inter-cellular signaling (e.g. if a protein is a ligand, a receptor, an extracellular matrix (ECM) component, etc.) This query type is very similar to annotations. However, <strong>intercell</strong> data does not come from original sources, but combined from several databases by us into categories (we also refer to the original sources).</p>
<p>We first inspect the different categories available in the OmniPath webserver. Then, we focus again in our previously selected complex and we check its the location of its individual components in the inter-cellular context. We can however see that the components of this particular complex are intracellular.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We check some of the different intercell categories</span>
<span class="fu"><a href="../reference/get_intercell_generic_categories.html">get_intercell_generic_categories</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "transmembrane"                       "transmembrane_predicted"            
##  [3] "peripheral"                          "plasma_membrane"                    
##  [5] "plasma_membrane_transmembrane"       "plasma_membrane_regulator"          
##  [7] "plasma_membrane_peripheral"          "secreted"                           
##  [9] "cell_surface"                        "ecm"                                
## [11] "ligand"                              "receptor"                           
## [13] "secreted_enzyme"                     "secreted_peptidase"                 
## [15] "extracellular"                       "intracellular"                      
## [17] "receptor_regulator"                  "secreted_receptor"                  
## [19] "sparc_ecm_regulator"                 "ecm_regulator"                      
## [21] "ligand_regulator"                    "cell_surface_ligand"                
## [23] "cell_adhesion"                       "matrix_adhesion"                    
## [25] "adhesion"                            "matrix_adhesion_regulator"          
## [27] "cell_surface_enzyme"                 "cell_surface_peptidase"             
## [29] "secreted_enyzme"                     "extracellular_peptidase"            
## [31] "secreted_peptidase_inhibitor"        "transporter"                        
## [33] "ion_channel"                         "ion_channel_regulator"              
## [35] "gap_junction"                        "tight_junction"                     
## [37] "adherens_junction"                   "desmosome"                          
## [39] "intracellular_intercellular_related"</code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We import the intercell data into a dataframe</span>
<span class="va">intercell</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_intercell.html">import_omnipath_intercell</a></span><span class="op">(</span>scope <span class="op">=</span> <span class="st">'generic'</span>,
    aspect <span class="op">=</span> <span class="st">'locational'</span><span class="op">)</span>

<span class="co">## We check the intercell annotations for the individual components of</span>
<span class="co">## our previous complex. We filter our data to print it in a good format</span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">intercell</span>,<span class="va">genesymbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">genes_complex</span><span class="op">)</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">genesymbol</span>, <span class="va">parent</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">category</span>, <span class="va">genesymbol</span>, <span class="va">parent</span><span class="op">)</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">genesymbol</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 4 x 3
##   category      genesymbol parent       
##   &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;        
## 1 intracellular PARP1      intracellular
## 2 intracellular WRN        intracellular
## 3 intracellular XRCC5      intracellular
## 4 intracellular XRCC6      intracellular</code></pre>
<p>The <code>import_intercell_network</code> function creates the most complete network, including many interactions which are false positives in the context of interacellular communication. It is highly recommended to apply some quality filtering on this network. The <code>high_confidence</code> parameter performs a quiet stringent filtering:</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">icn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_intercell_network.html">import_intercell_network</a></span><span class="op">(</span>high_confidence <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Using the function <code>filter_intercell_network</code> instead, you have much more flexibility to adjust the stringency of the filtering to the needs of your analysis. See the full list of options in the docs of the function.</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">icn</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/import_intercell_network.html">import_intercell_network</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://tidyr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="../reference/filter_intercell_network.html">filter_intercell_network</a></span><span class="op">(</span>
        min_curation_effort <span class="op">=</span> <span class="fl">1</span>,
        consensus_percentile <span class="op">=</span> <span class="fl">33</span>
    <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We close graphical connections</span>
<span class="kw">while</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.list</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>  <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="conclusion" class="section level1">
<h1 class="hasAnchor">
<a href="#conclusion" class="anchor" aria-hidden="true"></a>Conclusion</h1>
<p><em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> provides access to the wealth of data stored in the OmniPath webservice <a href="http://omnipathdb.org/" class="external-link uri">http://omnipathdb.org/</a> from the <em>R</em> enviroment. In addition, it contains some utility functions for visualization, filtering and analysis. The main strength of <em><a href="https://bioconductor.org/packages/3.13/OmnipathR" class="external-link">OmnipathR</a></em> is the straightforward transformation of the different OmniPath data into commonly used <em>R</em> objects, such as dataframes and graphs. Consequently, it allows an easy integration of the different types of data and a gateway to the vast number of <em>R</em> packages dedicated to the analysis and representaiton of biological data. We highlighted these abilities in some of the examples detailed in previous sections of this document.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor" aria-hidden="true"></a>Session info</h1>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Arch Linux
## 
## Matrix products: default
## BLAS:   /usr/lib/libopenblasp-r0.3.17.so
## LAPACK: /usr/lib/liblapack.so.3.10.0
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] gprofiler2_0.2.0 dnet_1.1.7       supraHex_1.26.0  hexbin_1.28.2   
## [5] igraph_1.2.6     tidyr_1.1.3      OmnipathR_3.1.2  BiocStyle_2.16.1
## 
## loaded via a namespace (and not attached):
##  [1] nlme_3.1-152        bitops_1.0-7        fs_1.5.0           
##  [4] bit64_4.0.5         progress_1.2.2      httr_1.4.2         
##  [7] rprojroot_2.0.2     Rgraphviz_2.32.0    tools_4.1.0        
## [10] backports_1.2.1     bslib_0.2.5.1       utf8_1.2.1         
## [13] R6_2.5.0            DBI_1.1.1           BiocGenerics_0.36.0
## [16] lazyeval_0.2.2      colorspace_2.0-2    tidyselect_1.1.1   
## [19] prettyunits_1.1.1   bit_4.0.4           curl_4.3.2         
## [22] compiler_4.1.0      textshaping_0.3.5   graph_1.66.0       
## [25] cli_3.0.1           xml2_1.3.2          desc_1.3.0         
## [28] plotly_4.9.4.1      bookdown_0.22       sass_0.4.0         
## [31] scales_1.1.1        checkmate_2.0.0     readr_2.0.0        
## [34] rappdirs_0.3.3      pkgdown_1.6.1.9001  systemfonts_1.0.2  
## [37] stringr_1.4.0       digest_0.6.27       rmarkdown_2.9.5    
## [40] pkgconfig_2.0.3     htmltools_0.5.1.1   fastmap_1.1.0      
## [43] htmlwidgets_1.5.3   rlang_0.4.11        readxl_1.3.1       
## [46] jquerylib_0.1.4     generics_0.1.0      jsonlite_1.7.2     
## [49] vroom_1.5.3         dplyr_1.0.7         RCurl_1.98-1.3     
## [52] magrittr_2.0.1      Matrix_1.3-4        Rcpp_1.0.7         
## [55] munsell_0.5.0       fansi_0.5.0         ape_5.5            
## [58] logger_0.2.1        lifecycle_1.0.0     stringi_1.7.3      
## [61] yaml_2.2.1          MASS_7.3-54         grid_4.1.0         
## [64] parallel_4.1.0      crayon_1.4.1        lattice_0.20-44    
## [67] hms_1.1.0           knitr_1.33          pillar_1.6.1       
## [70] stats4_4.1.0        glue_1.4.2          evaluate_0.14      
## [73] data.table_1.14.0   BiocManager_1.30.16 vctrs_0.3.8        
## [76] tzdb_0.1.2          cellranger_1.1.0    gtable_0.3.0       
## [79] purrr_0.3.4         assertthat_0.2.1    cachem_1.0.5       
## [82] ggplot2_3.3.5       xfun_0.24           later_1.2.0        
## [85] ragg_1.1.3          viridisLite_0.4.0   tibble_3.1.2       
## [88] memoise_2.0.0       ellipsis_0.3.2</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor" aria-hidden="true"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-BenShlomo2003" class="csl-entry">
Ben-Shlomo, I., S. Yu Hsu, R. Rauch, H. W. Kowalski, and A. J. W. Hsueh. 2003. <span>“Signaling Receptome: A Genomic and Evolutionary Perspective of Plasma Membrane Receptors Involved in Signal Transduction.”</span> <em>Science Signaling</em> 2003 (187): re9–9. <a href="https://doi.org/10.1126/stke.2003.187.re9" class="external-link">https://doi.org/10.1126/stke.2003.187.re9</a>.
</div>
<div id="ref-Berggrd2007" class="csl-entry">
Berggård, Tord, Sara Linse, and Peter James. 2007. <span>“Methods for the Detection and Analysis of Protein<span></span>protein Interactions.”</span> <em><span>PROTEOMICS</span></em> 7 (16): 2833–42. <a href="https://doi.org/10.1002/pmic.200700131" class="external-link">https://doi.org/10.1002/pmic.200700131</a>.
</div>
<div id="ref-Dinkel2010" class="csl-entry">
Dinkel, H., C. Chica, A. Via, C. M. Gould, L. J. Jensen, T. J. Gibson, and F. Diella. 2010. <span>“Phospho.<span>ELM</span>: A Database of Phosphorylation Sites–Update 2011.”</span> <em>Nucleic Acids Research</em> 39 (Database): D261–67. <a href="https://doi.org/10.1093/nar/gkq1104" class="external-link">https://doi.org/10.1093/nar/gkq1104</a>.
</div>
<div id="ref-Dobson2014" class="csl-entry">
Dobson, László, Tamás Langó, István Reményi, and Gábor E. Tusnády. 2014. <span>“Expediting Topology Data Gathering for the <span>TOPDB</span> Database.”</span> <em>Nucleic Acids Research</em> 43 (D1): D283–89. <a href="https://doi.org/10.1093/nar/gku1119" class="external-link">https://doi.org/10.1093/nar/gku1119</a>.
</div>
<div id="ref-Drew2017" class="csl-entry">
Drew, Kevin, Chanjae Lee, Ryan L Huizar, Fan Tu, Blake Borgeson, Claire D McWhite, Yun Ma, John B Wallingford, and Edward M Marcotte. 2017. <span>“Integration of over 9, 000 Mass Spectrometry Experiments Builds a Global Map of Human Protein Complexes.”</span> <em>Molecular Systems Biology</em> 13 (6): 932. <a href="https://doi.org/10.15252/msb.20167490" class="external-link">https://doi.org/10.15252/msb.20167490</a>.
</div>
<div id="ref-GarciaAlonso2019" class="csl-entry">
Garcia-Alonso, Luz, Christian H. Holland, Mahmoud M. Ibrahim, Denes Turei, and Julio Saez-Rodriguez. 2019. <span>“Benchmark and Integration of Resources for the Estimation of Human Transcription Factor Activities.”</span> <em>Genome Research</em> 29 (8): 1363–75. <a href="https://doi.org/10.1101/gr.240663.118" class="external-link">https://doi.org/10.1101/gr.240663.118</a>.
</div>
<div id="ref-Giurgiu2018" class="csl-entry">
Giurgiu, Madalina, Julian Reinhard, Barbara Brauner, Irmtraud Dunger-Kaltenbach, Gisela Fobo, Goar Frishman, Corinna Montrone, and Andreas Ruepp. 2018. <span>“<span>CORUM</span>: The Comprehensive Resource of Mammalian Protein Complexes<span></span>2019.”</span> <em>Nucleic Acids Research</em> 47 (D1): D559–63. <a href="https://doi.org/10.1093/nar/gky973" class="external-link">https://doi.org/10.1093/nar/gky973</a>.
</div>
<div id="ref-Hornbeck2014" class="csl-entry">
Hornbeck, Peter V., Bin Zhang, Beth Murray, Jon M. Kornhauser, Vaughan Latham, and Elzbieta Skrzypek. 2014. <span>“<span>PhosphoSitePlus</span>, 2014: Mutations, <span>PTMs</span> and Recalibrations.”</span> <em>Nucleic Acids Research</em> 43 (D1): D512–20. <a href="https://doi.org/10.1093/nar/gku1267" class="external-link">https://doi.org/10.1093/nar/gku1267</a>.
</div>
<div id="ref-Keerthikumar2016" class="csl-entry">
Keerthikumar, Shivakumar, David Chisanga, Dinuka Ariyaratne, Haidar Al Saffar, Sushma Anand, Kening Zhao, Monisha Samuel, et al. 2016. <span>“<span>ExoCarta</span>: A Web-Based Compendium of Exosomal Cargo.”</span> <em>Journal of Molecular Biology</em> 428 (4): 688–92. <a href="https://doi.org/10.1016/j.jmb.2015.09.019" class="external-link">https://doi.org/10.1016/j.jmb.2015.09.019</a>.
</div>
<div id="ref-Turei2016" class="csl-entry">
Türei, Dénes, Tamás Korcsmáros, and Julio Saez-Rodriguez. 2016. <span>“<span>OmniPath</span>: Guidelines and Gateway for Literature-Curated Signaling Pathway Resources.”</span> <em>Nature Methods</em> 13 (12): 966–67. <a href="https://doi.org/10.1038/nmeth.4077" class="external-link">https://doi.org/10.1038/nmeth.4077</a>.
</div>
<div id="ref-VentoTormo2018" class="csl-entry">
Vento-Tormo, Roser, Mirjana Efremova, Rachel A. Botting, Margherita Y. Turco, Miquel Vento-Tormo, Kerstin B. Meyer, Jong-Eun Park, et al. 2018. <span>“Single-Cell Reconstruction of the Early Maternal<span></span>fetal Interface in Humans.”</span> <em>Nature</em> 563 (7731): 347–53. <a href="https://doi.org/10.1038/s41586-018-0698-6" class="external-link">https://doi.org/10.1038/s41586-018-0698-6</a>.
</div>
</div>
</div>
  </div>
 <div class="col-md-3 d-none d-md-block" id="pkgdown-sidebar">
        <div class="sticky-top">
    <nav id="toc"><h2 data-toc-skip>Contents</h2>

    </nav>
</div>
      </div>

</div>



      <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alberto Valdeolivas, Denes Turei, Attila Gabor.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer>
</div>

  

  

  </body>
</html>
