<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>OmnipathR: utility functions to work with OmniPath in R • OmnipathR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="OmnipathR: utility functions to work with OmniPath in R">
<meta property="og:description" content="OmnipathR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OmnipathR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/OmnipathMainVignette.html">OmnipathR: utility functions to work with OmniPath in R</a>
    </li>
    <li>
      <a href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/OmnipathR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/omnipathdb">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://omnipathdb.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="OmnipathMainVignette_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>OmnipathR: utility functions to work with OmniPath in R</h1>
                        <h4 class="author">Alberto Valdeolivas</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:alvaldeolivas@gmail.com" class="email">alvaldeolivas@gmail.com</a>
      </address>
                              <h4 class="author">Attila Gabor</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:attila.gabor@bioquant.uni-heidelberg.de" class="email">attila.gabor@bioquant.uni-heidelberg.de</a>
      </address>
                              <h4 class="author">Denes Turei</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><h4 class="author">Julio Saez-Rodriguez</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/OmnipathR/blob/master/vignettes/OmnipathMainVignette.Rmd"><code>vignettes/OmnipathMainVignette.Rmd</code></a></small>
      <div class="hidden name"><code>OmnipathMainVignette.Rmd</code></div>

    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes how to use the <em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> package to retrieve information from the Omnipath database: <a href="http://omnipathdb.org/" class="uri">http://omnipathdb.org/</a>. In addition, it includes some utility functions to filter, analyse and visualize the data.</p>
    </div>
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> is an <em>R</em> package built to provide easy access to the data stored in the Omnipath webservice <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>:</p>
<p><a href="http://omnipathdb.org/" class="uri">http://omnipathdb.org/</a></p>
<p>The webservice implements a very simple REST style API. This package make requests by the HTTP protocol to retreive the data. Hence, fast Internet access is required for a proper use of <em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em>.</p>
<div id="query-types" class="section level2">
<h2 class="hasAnchor">
<a href="#query-types" class="anchor"></a>Query types</h2>
<p><em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> can retrieve five different types of data:</p>
<ul>
<li>
<p><strong>Interactions</strong>: protein-protein interactions organized in different datasets:</p>
<ul>
<li>
<strong>omnipath</strong>: the OmniPath data as defined in the original publication <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span> and collected from different databases.</li>
<li>
<strong>pathwayextra</strong>: activity flow interactions without literature reference.</li>
<li>
<strong>kinaseextra</strong>: enzyme-substrate interactions without literature reference.</li>
<li>
<strong>ligrecextra</strong>: ligand-receptor interactions without literature reference.</li>
<li>
<strong>tfregulons</strong>: transcription factor (TF)-target interactions from <strong>DoRothEA</strong> <span class="citation">(Garcia-Alonso et al. 2019)</span>.</li>
<li>
<strong>tf-miRNA</strong>: transcription factor-miRNA interactions</li>
<li>
<strong>miRNA-target</strong>: miRNA-mRNA interactions.</li>
<li>
<strong>lncRNA-mRNA</strong>: lncRNA-mRNA interactions.</li>
</ul>
</li>
<li><p><strong>Post-translational modifications (PTMs)</strong>: It provides enzyme-substrate reactions in a very similar way to the aforementioned interactions. Some of the biological databases related to PTMs integrated in Omnipath are Phospho.ELM <span class="citation">(Dinkel et al. 2010)</span> and PhosphoSitePlus [<span class="citation">Hornbeck et al. (2014)</span>}.</p></li>
<li><p><strong>Complexes</strong>: it provides access to a comprehensive database of more than 22000 protein complexes. This data comes from different resources such as: CORUM <span class="citation">(Giurgiu et al. 2018)</span> or Hu.map <span class="citation">(Drew et al. 2017)</span>.</p></li>
<li><p><strong>Annotations</strong>: it provides a large variety of data regarding different annotations about proteins and complexes. These data come from dozens of databases covering different topics such as: The Topology Data Bank of Transmembrane Proteins (TOPDB) <span class="citation">(Dobson et al. 2014)</span> or ExoCarta <span class="citation">(Keerthikumar et al. 2016)</span>, a database collecting the proteins that were identified in exosomes in multiple organisms.</p></li>
<li><p><strong>Intercell</strong>: it provides information on the roles in inter-cellular signaling. For instance. if a protein is a ligand, a receptor, an extracellular matrix (ECM) component, etc. The data does not come from original sources but combined from several databases by us. The source databases, such as CellPhoneDB <span class="citation">(Vento-Tormo et al. 2018)</span> or Receptome <span class="citation">(Ben-Shlomo et al. 2003)</span>, are also referred for each reacord.</p></li>
</ul>
<p>Figure @ref(fig:fig1) shows an overview of the resources featured in <strong>OmniPath</strong>. For more detailed information about the original data sources integrated in Omnipath, please visit:</p>
<ul>
<li><a href="http://omnipathdb.org/" class="uri">http://omnipathdb.org/</a></li>
<li><a href="http://omnipathdb.org/info" class="uri">http://omnipathdb.org/info</a></li>
</ul>
<div class="figure">
<img src="../man/figures/page1_1.png" alt=""><p class="caption">Overview of the resources featured in OmniPath. Causal resources (including activity-flow and enzyme-substrate resources) can provide direction (*) or sign and direction (+) of interactions.</p>
</div>
</div>
<div id="mouse-and-rat" class="section level2">
<h2 class="hasAnchor">
<a href="#mouse-and-rat" class="anchor"></a>Mouse and rat</h2>
<p>Excluding the miRNA interactions, all interactions and PTMs are available for human, mouse and rat. The rodent data has been translated from human using the NCBI Homologene database. Many human proteins do not have known homolog in rodents, hence rodent datasets are smaller than their human counterparts.</p>
<p>In case you work with mouse omics data you might do better to translate your dataset to human (for example using the pypath.homology module, <a href="https://github.com/saezlab/pypath/" class="uri">https://github.com/saezlab/pypath/</a>) and use human interaction data.</p>
</div>
</div>
<div id="installation-of-the-omnipathr-package" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-of-the-omnipathr-package" class="anchor"></a>Installation of the <em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> package</h1>
<p>First of all, you need a current version of <em>R</em>. <em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> is a freely available package deposited on <em>Bioconductor</em> and <a href="https://github.com/saezlab/OmnipathR">GitHub</a>. You can install it by running the following commands on an <em>R</em> console:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)

<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"OmnipathR"</span>)</pre></body></html></div>
<p>We also load here the required packages to run the code in this vignette.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">OmnipathR</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tidyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dnet</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">gprofiler2</span>)</pre></body></html></div>
</div>
<div id="usage-examples" class="section level1">
<h1 class="hasAnchor">
<a href="#usage-examples" class="anchor"></a>Usage Examples</h1>
<p>In the following paragraphs, we provide some examples to describe how to use the <em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> package to retrieve different types of information from <strong>Omnipath</strong> webserver. In addition, we play around with the data aiming at obtaining some biological relevant information.</p>
<p>Noteworthy, the sections <strong>complexes</strong>, <strong>annotations</strong> and <strong>intercell</strong> are linked. We explore the annotations and roles in inter-cellular communications of the proteins involved in a given complex. This basic example shows the usefulness of integrating the information available in the different <strong>Omnipath</strong> resources.</p>
<div id="interactions" class="section level2">
<h2 class="hasAnchor">
<a href="#interactions" class="anchor"></a>Interactions</h2>
<p>Proteins interact among them and with other biological molecules to perform cellular functions. Proteins also participates in pathways, linked series of reactions occurring inter/intra cells to transform products or to transmit signals inducing specific cellular responses. Protein interactions are therefore a very valuable source of information to understand cellular functioning.</p>
<p>We here download the original <strong>OmniPath</strong> human interactions <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>. To do so, we first check the different source databases and select some of them. Then, we print some of the downloaded interactions (“+” means activation, “-” means inhibition and “?” means undirected interactions or inconclusive data).</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co">## We check some of the different interaction databases</span>
<span class="fu"><a href="../reference/get_interaction_resources.html">get_interaction_resources</a></span>()</pre></body></html></div>
<pre><code>##   [1] "ABS"                       "ACSN"                     
##   [3] "Adhesome"                  "AlzPathway"               
##   [5] "ARACNe-GTEx_DoRothEA"      "ARN"                      
##   [7] "Baccin2019"                "BEL-Large-Corpus"         
##   [9] "BioGRID"                   "CA1"                      
##  [11] "CancerCellMap"             "CellPhoneDB"              
##  [13] "cellsignal.com"            "dbPTM"                    
##  [15] "DeathDomain"               "DEPOD"                    
##  [17] "Dinarello2013"             "DIP"                      
##  [19] "DOMINO"                    "DoRothEA_A"               
##  [21] "DoRothEA_B"                "DoRothEA_C"               
##  [23] "DoRothEA_D"                "DoRothEA-reviews_DoRothEA"
##  [25] "ELM"                       "EMBRACE"                  
##  [27] "ENCODE_tf-mirna"           "ENCODE-distal"            
##  [29] "ENCODE-proximal"           "FANTOM4_DoRothEA"         
##  [31] "Fantom5"                   "GO-lig-rec"               
##  [33] "Guide2Pharma"              "Guide2Pharma_CellPhoneDB" 
##  [35] "HOCOMOCO-v11_DoRothEA"     "HPMR"                     
##  [37] "HPRD"                      "HPRD-phos"                
##  [39] "HTRIdb"                    "HTRIdb_DoRothEA"          
##  [41] "I2D_CellPhoneDB"           "ICELLNET"                 
##  [43] "IMEx_CellPhoneDB"          "InnateDB"                 
##  [45] "InnateDB_CellPhoneDB"      "InnateDB-All_CellPhoneDB" 
##  [47] "IntAct_CellPhoneDB"        "IntAct_DoRothEA"          
##  [49] "iPTMnet"                   "iTALK"                    
##  [51] "JASPAR-v2018_DoRothEA"     "KEA"                      
##  [53] "KEGG"                      "KEGG_DoRothEA"            
##  [55] "Kinexus"                   "Kirouac2010"              
##  [57] "Li2012"                    "LMPID"                    
##  [59] "lncrnadb"                  "LncRNADisease"            
##  [61] "LRdb"                      "Macrophage"               
##  [63] "MatrixDB"                  "MatrixDB_CellPhoneDB"     
##  [65] "MIMP"                      "MINT_CellPhoneDB"         
##  [67] "miR2Disease"               "miRDeathDB"               
##  [69] "miRecords"                 "miRTarBase"               
##  [71] "MPPI"                      "NCI-PID"                  
##  [73] "ncRDeathDB"                "NetPath"                  
##  [75] "NetworKIN"                 "NFIRegulomeDB_DoRothEA"   
##  [77] "NRF2ome"                   "ORegAnno"                 
##  [79] "ORegAnno_DoRothEA"         "PAZAR"                    
##  [81] "PAZAR_DoRothEA"            "PDZBase"                  
##  [83] "phosphoELM"                "PhosphoNetworks"          
##  [85] "PhosphoPoint"              "PhosphoSite"              
##  [87] "PhosphoSite_noref"         "ProtMapper"               
##  [89] "Ramilowski2015"            "REACH"                    
##  [91] "Reactome"                  "RegNetwork_DoRothEA"      
##  [93] "ReMap_DoRothEA"            "RLIMS-P"                  
##  [95] "SignaLink3"                "SIGNOR"                   
##  [97] "Sparser"                   "SPIKE"                    
##  [99] "STRING"                    "TCRcuration"              
## [101] "TFactS_DoRothEA"           "TFe_DoRothEA"             
## [103] "TransmiR"                  "TRED_DoRothEA"            
## [105] "TRIP"                      "TRRD_DoRothEA"            
## [107] "TRRUST_DoRothEA"           "UniProt"                  
## [109] "UniProt_CellPhoneDB"       "Wang"</code></pre>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co">## The interactions are stored into a data frame.</span>
<span class="no">interactions</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"SignaLink3"</span>,<span class="st">"PhosphoSite"</span>,
    <span class="st">"SIGNOR"</span>))</pre></body></html></div>
<pre><code>## Downloaded 17420 interactions.</code></pre>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## We visualize the first interactions in the data frame.</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">interactions</span>))</pre></body></html></div>
<pre><code>##            source interaction         target n_resources n_references
## 4    SRC (P12931)  ==( + )==&gt; TRPV1 (Q8NER1)           9            6
## 2  PRKG1 (Q13976)  ==( - )==&gt; TRPC6 (Q9Y210)           7            5
## 1  PRKG1 (Q13976)  ==( - )==&gt; TRPC3 (Q13507)           9            2
## 5    LYN (P07948)  ==( + )==&gt; TRPV4 (Q9HBA0)           9            2
## 6  PTPN1 (P18031)  ==( - )==&gt; TRPV6 (Q9H1D0)           3            2
## 3 PRKACA (P17612)  ==( + )==&gt; TRPV1 (Q8NER1)           6            1</code></pre>
<div id="protein-protein-interaction-networks" class="section level3">
<h3 class="hasAnchor">
<a href="#protein-protein-interaction-networks" class="anchor"></a>Protein-protein interaction networks</h3>
<p>Protein-protein interactions are usually converted into networks. Describing protein interactions as networks not only provides a convenient format for visualization, but also allows applying graph theory methods to mine the biological information they contain.</p>
<p>We convert here our set of interactions to a network/graph (<em><a href="https://CRAN.R-project.org/package=igraph">igraph</a></em> object). Then, we apply two very common approaches to extract information from a biological network:</p>
<ul>
<li>
<strong>Shortest Paths</strong>: finding a path between two nodes (proteins) going through the minimum number of edges. This can be very useful to track consecutive reactions within a given pathway. We display below the shortest path between two given proteins and all the possible shortests paths between two other proteins. It is to note that the functions <code>printPath\_es</code> and <code>printPath\_vs</code> display very similar results, but the first one takes as an input an edge sequence and the second one a node sequence.</li>
</ul>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co">## We transform the interactions data frame into a graph</span>
<span class="no">OPI_g</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span>(<span class="kw">interactions</span> <span class="kw">=</span> <span class="no">interactions</span>)

<span class="co">## Find and print shortest paths on the directed network between proteins</span>
<span class="co">## of interest:</span>
<span class="fu"><a href="../reference/printPath_es.html">printPath_es</a></span>(<span class="fu">shortest_paths</span>(<span class="no">OPI_g</span>,<span class="kw">from</span> <span class="kw">=</span> <span class="st">"TYRO3"</span>,<span class="kw">to</span> <span class="kw">=</span> <span class="st">"STAT3"</span>,
    <span class="kw">output</span> <span class="kw">=</span> <span class="st">'epath'</span>)$<span class="no">epath</span><span class="kw">[[</span><span class="fl">1</span>]],<span class="no">OPI_g</span>)</pre></body></html></div>
<pre><code>##            source interaction          target n_resources n_references
## 1  TYRO3 (Q06418)  ==( ? )==&gt;   AKT1 (P31749)           2            0
## 2   AKT1 (P31749)  ==( - )==&gt; DAB2IP (Q5VWQ8)           3            1
## 3 DAB2IP (Q5VWQ8)  ==( - )==&gt;  STAT3 (P40763)           1            1</code></pre>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co">## Find and print all shortest paths between proteins of interest:</span>
<span class="fu"><a href="../reference/printPath_vs.html">printPath_vs</a></span>(<span class="fu">all_shortest_paths</span>(<span class="no">OPI_g</span>,<span class="kw">from</span> <span class="kw">=</span> <span class="st">"DYRK2"</span>,
    <span class="kw">to</span> <span class="kw">=</span> <span class="st">"MAPKAPK2"</span>)$<span class="no">res</span>,<span class="no">OPI_g</span>)</pre></body></html></div>
<pre><code>## Pathway 1: DYRK2 -&gt; TBK1 -&gt; NFKB1 -&gt; MAP3K8 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 2: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP3K8 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 3: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP3K8 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 4: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP3K8 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 5: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PEA15 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 6: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PEA15 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 7: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PEA15 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 8: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PPP1CA -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 9: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PPP1CA -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 10: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PPP1CA -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 11: DYRK2 -&gt; TBK1 -&gt; PEBP1 -&gt; MAP2K1 -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 12: DYRK2 -&gt; TP53 -&gt; RPS6KA5 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 13: DYRK2 -&gt; TP53 -&gt; RPS6KA4 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 14: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 15: DYRK2 -&gt; TP53 -&gt; RPS6KA2 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 16: DYRK2 -&gt; TP53 -&gt; RPS6KA1 -&gt; CDC25A -&gt; MAPK3 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 17: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP2K4 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 18: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP2K4 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 19: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP2K4 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 20: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; MAP3K5 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 21: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP3K5 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 22: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP3K5 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 23: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP3K5 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 24: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; RAC1 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 25: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; RAC1 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 26: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; RAC1 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 27: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; SH3RF1 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 28: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; SH3RF1 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 29: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; SH3RF1 -&gt; MAPK8 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 30: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP2K4 -&gt; MAPK10 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 31: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP2K4 -&gt; MAPK10 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 32: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP2K4 -&gt; MAPK10 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 33: DYRK2 -&gt; TP53 -&gt; RPS6KA5 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 34: DYRK2 -&gt; TP53 -&gt; RPS6KA4 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 35: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 36: DYRK2 -&gt; TP53 -&gt; RPS6KA2 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 37: DYRK2 -&gt; TP53 -&gt; RPS6KA1 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 38: DYRK2 -&gt; TBK1 -&gt; RPS6KB1 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 39: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 40: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 41: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 42: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP2K4 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 43: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP2K4 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 44: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP2K4 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 45: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 46: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 47: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 48: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 49: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; CSNK2A1 -&gt; MAPK9 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 50: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP2K4 -&gt; MAPK9 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 51: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP2K4 -&gt; MAPK9 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 52: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP2K4 -&gt; MAPK9 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 53: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; RAC1 -&gt; MAPK9 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 54: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; RAC1 -&gt; MAPK9 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 55: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; RAC1 -&gt; MAPK9 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 56: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; CSNK2A1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 57: DYRK2 -&gt; TBK1 -&gt; NFKB1 -&gt; MAP3K8 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 58: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; MAP3K8 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 59: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; MAP3K8 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 60: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP3K8 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 61: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PEA15 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 62: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PEA15 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 63: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PEA15 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 64: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; PPP1CA -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 65: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; PPP1CA -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 66: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PPP1CA -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 67: DYRK2 -&gt; TBK1 -&gt; PEBP1 -&gt; RAF1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 68: DYRK2 -&gt; TBK1 -&gt; AKT3 -&gt; RAF1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 69: DYRK2 -&gt; TBK1 -&gt; AKT2 -&gt; RAF1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 70: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; RAF1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 71: DYRK2 -&gt; TBK1 -&gt; PEBP1 -&gt; MAP2K1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 72: DYRK2 -&gt; TP53 -&gt; RPS6KA3 -&gt; DAPK1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 73: DYRK2 -&gt; TP53 -&gt; RPS6KA2 -&gt; DAPK1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<pre><code>## Pathway 74: DYRK2 -&gt; TP53 -&gt; RPS6KA1 -&gt; DAPK1 -&gt; MAPK1 -&gt; MAPKAPK2</code></pre>
<ul>
<li>
<strong>Clustering</strong>: grouping nodes (proteins) in such a way that nodes belonging to the same group (called cluster) are more connected in the network to each other than to those in other groups (clusters). Since proteins interact to perform their functions, proteins within the same cluster are likely to be implicated in similar biological tasks. Figure @ref(fig:fig2) shows the subgraph containing the proteins and interactions of a specifc protein, CD22. The <em><a href="https://CRAN.R-project.org/package=igraph">igraph</a></em> package contains functions to apply sevaral different cluster methods on graphs (visit <a href="https://igraph.org/r/doc/" class="uri">https://igraph.org/r/doc/</a> for detailed information.)</li>
</ul>
<div class="sourceCode" id="cb86"><html><body><pre class="r"><span class="co">## We apply a clustering algorithm (Louvain) to group proteins in</span>
<span class="co">## our network. We apply here Louvain which is fast but can only run</span>
<span class="co">## on undirected graphs. Other clustering algorithms can deal with</span>
<span class="co">## directed networks but with longer computational times,</span>
<span class="co">## such as cluster_edge_betweenness. These cluster methods are directly</span>
<span class="co">## available in the igraph package.</span>
<span class="no">OPI_g_undirected</span> <span class="kw">&lt;-</span> <span class="fu">as.undirected</span>(<span class="no">OPI_g</span>, <span class="kw">mode</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mutual"</span>))
<span class="no">OPI_g_undirected</span> <span class="kw">&lt;-</span> <span class="fu">simplify</span>(<span class="no">OPI_g_undirected</span>)
<span class="no">cl_results</span> <span class="kw">&lt;-</span> <span class="fu">cluster_fast_greedy</span>(<span class="no">OPI_g_undirected</span>)
<span class="co">## We extract the cluster where a protein of interest is contained</span>
<span class="no">cluster_id</span> <span class="kw">&lt;-</span> <span class="no">cl_results</span>$<span class="no">membership</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">cl_results</span>$<span class="no">names</span> <span class="kw">==</span> <span class="st">"SOCS3"</span>)]
<span class="no">module_graph</span> <span class="kw">&lt;-</span> <span class="fu">induced_subgraph</span>(<span class="no">OPI_g_undirected</span>,
    <span class="fu">V</span>(<span class="no">OPI_g</span>)$<span class="no">name</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">cl_results</span>$<span class="no">membership</span> <span class="kw">==</span> <span class="no">cluster_id</span>)])</pre></body></html></div>
<div class="figure">
<img src="OmnipathMainVignette_files/figure-html/fig2-1.png" alt="CD22 associated cluser. Subnetwork extracted from the interactions graph representing the cluster where we can find the gene *CD22* (yellow node)" width="3000"><p class="caption">
CD22 associated cluser. Subnetwork extracted from the interactions graph representing the cluster where we can find the gene <em>CD22</em> (yellow node)
</p>
</div>
</div>
</div>
<div id="other-interaction-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#other-interaction-datasets" class="anchor"></a>Other interaction datasets</h2>
<p>We used above the interactions from the dataset described in the original <strong>Omnipath</strong> publication <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>. In this section, we provide examples on how to retry and deal with interactions from the remaining datasets. The same functions can been applied to every interaction dataset.</p>
<div id="pathway-extra" class="section level3">
<h3 class="hasAnchor">
<a href="#pathway-extra" class="anchor"></a>Pathway Extra</h3>
<p>In the first example, we are going to get the interactions from the <strong>pathwayextra</strong> dataset, which contains activity flow interactions without literature reference. We are going to focus on the mouse interactions for a given gene in this particular case.</p>
<div class="sourceCode" id="cb87"><html><body><pre class="r"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="no">interactions</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/import_pathwayextra_interactions.html">import_pathwayextra_interactions</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"BioGRID"</span>,<span class="st">"STRING"</span>),
    <span class="kw">organism</span> <span class="kw">=</span> <span class="fl">10090</span>)</pre></body></html></div>
<pre><code>## Downloaded 1028 interactions.</code></pre>
<div class="sourceCode" id="cb89"><html><body><pre class="r"><span class="co">## We select all the interactions in which Amfr gene is involved</span>
<span class="no">interactions_Amfr</span> <span class="kw">&lt;-</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">interactions</span>, <span class="no">source_genesymbol</span> <span class="kw">==</span> <span class="st">"Amfr"</span> <span class="kw">|</span>
    <span class="no">target_genesymbol</span> <span class="kw">==</span> <span class="st">"Amfr"</span>)

<span class="co">## We print these interactions:</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span>(<span class="no">interactions_Amfr</span>)</pre></body></html></div>
<pre><code>##          source interaction       target n_resources n_references
## 1 Amfr (Q9R049)  ==( + )==&gt; Vcp (Q01853)           3            5</code></pre>
</div>
<div id="kinase-extra" class="section level3">
<h3 class="hasAnchor">
<a href="#kinase-extra" class="anchor"></a>Kinase Extra</h3>
<p>Next, we download the interactions from the <strong>kinaseextra</strong> dataset, which contains enzyme-substrate interactions without literature reference. We are going to focus on rat reactions targeting a particular gene.</p>
<div class="sourceCode" id="cb91"><html><body><pre class="r"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="no">interactions</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/import_kinaseextra_interactions.html">import_kinaseextra_interactions</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"PhosphoPoint"</span>,
    <span class="st">"PhosphoSite"</span>), <span class="kw">organism</span> <span class="kw">=</span> <span class="fl">10116</span>)</pre></body></html></div>
<pre><code>## Downloaded 4937 interactions.</code></pre>
<div class="sourceCode" id="cb93"><html><body><pre class="r"><span class="co">## We select the interactions in which Dpysl2 gene is a target</span>
<span class="no">interactions_TargetDpysl2</span> <span class="kw">&lt;-</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">interactions</span>,
    <span class="no">target_genesymbol</span> <span class="kw">==</span> <span class="st">"Dpysl2"</span>)

<span class="co">## We print these interactions:</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span>(<span class="no">interactions_TargetDpysl2</span>)</pre></body></html></div>
<pre><code>##           source interaction          target n_resources n_references
## 1 Gsk3b (P18266)  ==(+/-)==&gt; Dpysl2 (P47942)          18           32
## 4  Cdk5 (Q03114)  ==( + )==&gt; Dpysl2 (P47942)          11           26
## 2 Rock2 (Q62868)  ==( + )==&gt; Dpysl2 (P47942)          12            6
## 3   Fer (P09760)  ==( ? )==&gt; Dpysl2 (P47942)           4            2
## 6 Rock1 (Q63644)  ==( ? )==&gt; Dpysl2 (P47942)           9            1
## 5 Gsk3a (P18265)  ==( ? )==&gt; Dpysl2 (P47942)           6            1</code></pre>
</div>
<div id="ligand-receptor-extra" class="section level3">
<h3 class="hasAnchor">
<a href="#ligand-receptor-extra" class="anchor"></a>Ligand-receptor Extra</h3>
<p>In the following example we are going to work with the <strong>ligrecextra</strong> dataset, which contains ligand-receptor interactions without literature reference. Our goal is to find the potential receptors associated to a given ligand, CDH1 (Figure @ref(fig:fig3)).</p>
<div class="sourceCode" id="cb95"><html><body><pre class="r"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="no">interactions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_ligrecextra_interactions.html">import_ligrecextra_interactions</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"HPRD"</span>,
    <span class="st">"Guide2Pharma"</span>), <span class="kw">organism</span><span class="kw">=</span><span class="fl">9606</span>)</pre></body></html></div>
<pre><code>## Downloaded 1512 interactions.</code></pre>
<div class="sourceCode" id="cb97"><html><body><pre class="r"><span class="co">## Receptors of the CDH1 ligand.</span>
<span class="no">interactions_CDH1</span> <span class="kw">&lt;-</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">interactions</span>, <span class="no">source_genesymbol</span> <span class="kw">==</span> <span class="st">"CDH1"</span>)

<span class="co">## We transform the interactions data frame into a graph</span>
<span class="no">OPI_g</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span>(<span class="kw">interactions</span> <span class="kw">=</span> <span class="no">interactions_CDH1</span>)

<span class="co">## We induce a network with these genes</span>
<span class="no">Induced_Network</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/pkg/dnet/man/dNetInduce.html">dNetInduce</a></span>(<span class="kw">g</span><span class="kw">=</span><span class="no">OPI_g</span>,
    <span class="kw">nodes_query</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>( <span class="fu">V</span>(<span class="no">OPI_g</span>)$<span class="no">name</span>), <span class="kw">knn</span><span class="kw">=</span><span class="fl">0</span>,
    <span class="kw">remove.loops</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">largest.comp</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<div class="figure">
<img src="OmnipathMainVignette_files/figure-html/fig3-1.png" alt="Ligand-receptor interactions for the CDH1 ligand." width="2187.5"><p class="caption">
Ligand-receptor interactions for the CDH1 ligand.
</p>
</div>
</div>
<div id="dorothea-regulons" class="section level3">
<h3 class="hasAnchor">
<a href="#dorothea-regulons" class="anchor"></a>DoRothEA Regulons</h3>
<p>Another very interesting interaction dataset also available in OmniPath is <strong>DoRothEA</strong> <span class="citation">(Garcia-Alonso et al. 2019)</span>. It contains transcription factor (TF)-target interactions with confidence score, ranging from A-E, being A the most confident interactions. In the code chunk shown below, we select and print the most confident interactions for a given TF.</p>
<div class="sourceCode" id="cb98"><html><body><pre class="r"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="no">interactions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_dorothea_interactions.html">import_dorothea_interactions</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"DoRothEA_A"</span>,
    <span class="st">"ARACNe-GTEx_DoRothEA"</span>), <span class="kw">organism</span><span class="kw">=</span><span class="fl">9606</span>)</pre></body></html></div>
<pre><code>## Downloaded 6976 interactions.</code></pre>
<div class="sourceCode" id="cb100"><html><body><pre class="r"><span class="co">## We select the most confident interactions for a given TF and we print</span>
<span class="co">## the interactions to check the way it regulates its different targets</span>
<span class="no">interactions_A_GLI1</span>  <span class="kw">&lt;-</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">interactions</span>, <span class="no">dorothea_level</span><span class="kw">==</span><span class="st">"A"</span>,
    <span class="no">source_genesymbol</span> <span class="kw">==</span> <span class="st">"GLI1"</span>)
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span>(<span class="no">interactions_A_GLI1</span>)</pre></body></html></div>
<pre><code>##          source interaction          target n_resources n_references
## 1 GLI1 (P08151)  ==( + )==&gt;  PTCH1 (Q13635)           1            3
## 2 GLI1 (P08151)  ==( + )==&gt;   BCL2 (P10415)           0            3
## 5 GLI1 (P08151)  ==( + )==&gt;  SFRP1 (Q8N474)           0            2
## 3 GLI1 (P08151)  ==( - )==&gt;   EGR2 (P11161)           0            1
## 4 GLI1 (P08151)  ==( + )==&gt; IGFBP6 (P24592)           0            1
## 6 GLI1 (P08151)  ==( - )==&gt;  SLIT2 (O94813)           0            1</code></pre>
</div>
<div id="mirna-target-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#mirna-target-dataset" class="anchor"></a>miRNA-target dataset</h3>
<p>The last dataset describing interactions is <strong>mirnatarget</strong>. It stores miRNA-mRNA and TF-miRNA interactions. These interactions are only available for human so far. We next select the miRNA interacting with the TF selected in the previous code chunk, <em>GLI1</em>. The main function of miRNAs seems to be related with gene regulation. It is therefore interesting to see how some miRNA can regulate the expression of a TF which in turn regulates the expression of other genes. Figure @ref(fig:fig4) shows a schematic network of the miRNA targeting <em>GLI1</em> and the genes regulated by this TF.</p>
<div class="sourceCode" id="cb102"><html><body><pre class="r"><span class="co">## We query and store the interactions into a dataframe</span>
<span class="no">interactions</span> <span class="kw">&lt;-</span>
  <span class="fu"><a href="../reference/import_mirnatarget_interactions.html">import_mirnatarget_interactions</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"miRTarBase"</span>,<span class="st">"miRecords"</span>))</pre></body></html></div>
<pre><code>## Downloaded 5962 interactions.</code></pre>
<div class="sourceCode" id="cb104"><html><body><pre class="r"><span class="co">## We select the interactions where a miRNA is interacting with the TF</span>
<span class="co">## used in the previous code chunk and we print these interactions.</span>
<span class="no">interactions_miRNA_GLI1</span> <span class="kw">&lt;-</span>
    <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">interactions</span>,  <span class="no">target_genesymbol</span> <span class="kw">==</span> <span class="st">"GLI1"</span>)
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span>(<span class="no">interactions_miRNA_GLI1</span>)</pre></body></html></div>
<pre><code>##                          source interaction        target n_resources
## 2 hsa-miR-324-5p (MIMAT0000761)  ==( ? )==&gt; GLI1 (P08151)           3
## 1   hsa-miR-125b (MIMAT0000423)  ==( ? )==&gt; GLI1 (P08151)           2
## 3    hsa-miR-326 (MIMAT0000756)  ==( ? )==&gt; GLI1 (P08151)           2
## 4 hsa-miR-202-3p (MIMAT0002811)  ==( ? )==&gt; GLI1 (P08151)           1
## 5   hsa-miR-133b (MIMAT0000770)  ==( ? )==&gt; GLI1 (P08151)           1
##   n_references
## 2            2
## 1            1
## 3            1
## 4            1
## 5            1</code></pre>
<div class="sourceCode" id="cb106"><html><body><pre class="r"><span class="co">## We transform the previous selections to graphs (igraph objects)</span>
<span class="no">OPI_g_1</span> <span class="kw">&lt;-</span><span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span>(<span class="kw">interactions</span> <span class="kw">=</span> <span class="no">interactions_A_GLI1</span>)
<span class="no">OPI_g_2</span> <span class="kw">&lt;-</span><span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span>(<span class="kw">interactions</span> <span class="kw">=</span> <span class="no">interactions_miRNA_GLI1</span>)</pre></body></html></div>
<div class="figure">
<img src="OmnipathMainVignette_files/figure-html/fig4-1.png" alt="miRNA-TF-target network. Schematic network of the miRNA (red square nodes) targeting  extit{GLI1} (yellow node) and the genes regulated by this TF (blue round nodes)." width="2187.5"><p class="caption">
miRNA-TF-target network. Schematic network of the miRNA (red square nodes) targeting extit{GLI1} (yellow node) and the genes regulated by this TF (blue round nodes).
</p>
</div>
</div>
</div>
<div id="post-translational-modifications-ptms" class="section level2">
<h2 class="hasAnchor">
<a href="#post-translational-modifications-ptms" class="anchor"></a>Post-translational modifications (PTMs)</h2>
<p>Another query type available is PTMs which provides enzyme-substrate reactions in a very similar way to the aforementioned interactions. PTMs refer generally to enzymatic modification of proteins after their synthesis in the ribosomes. PTMs can be highly context-specific and they play a main role in the activation/inhibition of biological pathways.</p>
<p>In the next code chunk, we download the <strong>PTMs</strong> for human. We first check the different available source databases, even though we do not perform any filter. Then, we select and print the reactions involving a specific enzyme-substrate pair. Those reactions lack information about activation or inhibition. To obtain that information, we match the data with <strong>OmniPath</strong> interactions. Finally, we show that it is also possible to build a graph using this information, and to retrieve PTMs from mouse or rat.</p>
<div class="sourceCode" id="cb107"><html><body><pre class="r"><span class="co">## We check the different PTMs databases</span>
<span class="fu"><a href="../reference/get_enzsub_resources.html">get_enzsub_resources</a></span>()</pre></body></html></div>
<pre><code>##  [1] "BEL-Large-Corpus_ProtMapper" "dbPTM"                      
##  [3] "DEPOD"                       "HPRD"                       
##  [5] "HPRD_MIMP"                   "KEA"                        
##  [7] "Li2012"                      "MIMP"                       
##  [9] "NCI-PID_ProtMapper"          "phosphoELM"                 
## [11] "phosphoELM_MIMP"             "PhosphoNetworks"            
## [13] "PhosphoSite"                 "PhosphoSite_MIMP"           
## [15] "PhosphoSite_ProtMapper"      "ProtMapper"                 
## [17] "REACH_ProtMapper"            "Reactome_ProtMapper"        
## [19] "RLIMS-P_ProtMapper"          "SIGNOR"                     
## [21] "SIGNOR_ProtMapper"           "Sparser_ProtMapper"</code></pre>
<div class="sourceCode" id="cb109"><html><body><pre class="r"><span class="co">## We query and store the ptms into a dataframe. No filtering by</span>
<span class="co">## databases in this case.</span>
<span class="no">ptms</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_enzsub.html">import_omnipath_enzsub</a></span>()</pre></body></html></div>
<pre><code>## Downloaded 39639 enzyme-substrate relationships.</code></pre>
<div class="sourceCode" id="cb111"><html><body><pre class="r"><span class="co">## We can select and print the reactions between a specific kinase and</span>
<span class="co">## a specific substrate</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span>(<span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">ptms</span>,<span class="no">enzyme_genesymbol</span><span class="kw">==</span><span class="st">"MAP2K1"</span>,
    <span class="no">substrate_genesymbol</span><span class="kw">==</span><span class="st">"MAPK3"</span>))</pre></body></html></div>
<pre><code>##            enzyme interaction           substrate    modification n_resources
## 1 MAP2K1 (Q02750)       ====&gt; MAPK3_Y204 (P27361) phosphorylation           8
## 2 MAP2K1 (Q02750)       ====&gt; MAPK3_T202 (P27361) phosphorylation           8
## 3 MAP2K1 (Q02750)       ====&gt; MAPK3_Y210 (P27361) phosphorylation           2
## 4 MAP2K1 (Q02750)       ====&gt; MAPK3_T207 (P27361) phosphorylation           2
## 5 MAP2K1 (Q02750)       ====&gt;  MAPK3_T80 (P27361) phosphorylation           1
## 6 MAP2K1 (Q02750)       ====&gt; MAPK3_Y222 (P27361) phosphorylation           1</code></pre>
<div class="sourceCode" id="cb113"><html><body><pre class="r"><span class="co">## In the previous results, we can see that ptms does not contain sign</span>
<span class="co">## (activation/inhibition). We can generate this information based on the</span>
<span class="co">## protein-protein Omnipath interaction dataset.</span>
<span class="no">interactions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions</a></span>()</pre></body></html></div>
<pre><code>## Downloaded 36684 interactions.</code></pre>
<div class="sourceCode" id="cb115"><html><body><pre class="r"><span class="no">ptms</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_signed_ptms.html">get_signed_ptms</a></span>(<span class="no">ptms</span>, <span class="no">interactions</span>)

<span class="co">## We select again the same kinase and substrate. Now we have information</span>
<span class="co">## about inhibition or activation when we print the ptms</span>
<span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span>(<span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">ptms</span>,<span class="no">enzyme_genesymbol</span><span class="kw">==</span><span class="st">"MAP2K1"</span>,
    <span class="no">substrate_genesymbol</span><span class="kw">==</span><span class="st">"MAPK3"</span>))</pre></body></html></div>
<pre><code>##            enzyme interaction           substrate    modification n_resources
## 6 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y204 (P27361) phosphorylation           8
## 4 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_T202 (P27361) phosphorylation           8
## 2 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y210 (P27361) phosphorylation           2
## 3 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_T207 (P27361) phosphorylation           2
## 1 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y222 (P27361) phosphorylation           1
## 5 MAP2K1 (Q02750)  ==( + )==&gt;  MAPK3_T80 (P27361) phosphorylation           1</code></pre>
<div class="sourceCode" id="cb117"><html><body><pre class="r"><span class="co">## We can also transform the ptms into a graph.</span>
<span class="no">ptms_g</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ptms_graph.html">ptms_graph</a></span>(<span class="kw">ptms</span> <span class="kw">=</span> <span class="no">ptms</span>)

<span class="co">## We download PTMs for mouse</span>
<span class="no">ptms</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_enzsub.html">import_omnipath_enzsub</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"PhosphoSite"</span>, <span class="st">"SIGNOR"</span>),
    <span class="kw">organism</span><span class="kw">=</span><span class="fl">10090</span>)</pre></body></html></div>
<pre><code>## Downloaded 14136 enzyme-substrate relationships.</code></pre>
</div>
<div id="complexes" class="section level2">
<h2 class="hasAnchor">
<a href="#complexes" class="anchor"></a>Complexes</h2>
<p>Some studies indicate that around 80% of the human proteins operate in complexes, and many proteins belong to several different complexes <span class="citation">(Berggård, Linse, and James 2007)</span>. These complexes play critical roles in a large variety of biological processes. Some well-known examples are the proteasome and the ribosome. Thus, the description of the full set of protein complexes functioning in cells is essential to improve our understanding of biological processes.</p>
<p>The <strong>complexes</strong> query provides access to more than 20000 protein complexes. This comprehensive database has been created by integrating different resources. We now download these molecular complexes filtering by some of the source databases. We check the complexes where a couple of specific genes participate. First, we look for the complexes where any of these two genes participate. We then identify the complex where these two genes are jointly involved. Finally, we perform an enrichment analysis with the genes taking part in that complex. You should keep an eye on this complex since it will be used again in the forthcoming sections.</p>
<div class="sourceCode" id="cb119"><html><body><pre class="r"><span class="co">## We check the different complexes databases</span>
<span class="fu"><a href="../reference/get_complex_resources.html">get_complex_resources</a></span>()</pre></body></html></div>
<pre><code>##  [1] "CellPhoneDB"    "CFinder"        "Compleat"       "ComplexPortal" 
##  [5] "CORUM"          "Guide2Pharma"   "Havugimana2012" "HPMR"          
##  [9] "hu.MAP"         "ICELLNET"       "NetworkBlast"   "PDB"           
## [13] "Signor"         "SIGNOR"</code></pre>
<div class="sourceCode" id="cb121"><html><body><pre class="r"><span class="co">## We query and store complexes from some sources into a dataframe.</span>
<span class="no">complexes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_complexes.html">import_omnipath_complexes</a></span>(<span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CORUM"</span>, <span class="st">"hu.MAP"</span>))</pre></body></html></div>
<pre><code>## Downloaded 7215 protein complexes.</code></pre>
<div class="sourceCode" id="cb123"><html><body><pre class="r"><span class="co">## We check all the molecular complexes where a set of genes participate</span>
<span class="no">query_genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"WRN"</span>,<span class="st">"PARP1"</span>)

<span class="co">## Complexes where any of the input genes participate</span>
<span class="no">complexes_query_genes_any</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="../reference/get_complex_genes.html">get_complex_genes</a></span>(<span class="no">complexes</span>,<span class="no">query_genes</span>,
    <span class="kw">total_match</span><span class="kw">=</span><span class="fl">FALSE</span>))

<span class="co">## We print the components of the different selected components</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">complexes_query_genes_any</span>$<span class="no">components_genesymbols</span>,<span class="fl">6</span>)</pre></body></html></div>
<pre><code>## [1] "NCAPD2_NCAPG_NCAPH_PARP1_SMC2_SMC4_XRCC1"                             
## [2] "CCNA2_CDK2_LIG1_PARP1_POLA1_POLD1_POLE_RFC1_RFC2_RPA1_RPA2_RPA3_TOP1" 
## [3] "CCNA2_CCNB1_CDK1_PARP1_POLA1_POLD1_POLE_RFC1_RFC2_RPA1_RPA2_RPA3_TOP1"
## [4] "MRE11_PARP1_RAD50_TERF2_TERF2IP_XRCC5_XRCC6"                          
## [5] "TERF2_WRN"                                                            
## [6] "CALR_DHX30_H2AFX_HIST3H2BB_HSPA5_NPM1_PARP1"</code></pre>
<div class="sourceCode" id="cb125"><html><body><pre class="r"><span class="co">## Complexes where all the input genes participate jointly</span>
<span class="no">complexes_query_genes_join</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="../reference/get_complex_genes.html">get_complex_genes</a></span>(<span class="no">complexes</span>,<span class="no">query_genes</span>,
    <span class="kw">total_match</span><span class="kw">=</span><span class="fl">TRUE</span>))

<span class="co">## We print the components of the different selected components</span>
<span class="no">complexes_query_genes_join</span>$<span class="no">components_genesymbols</span></pre></body></html></div>
<pre><code>## [1] "PARP1_WRN_XRCC5_XRCC6"</code></pre>
<div class="sourceCode" id="cb127"><html><body><pre class="r"><span class="no">genes_complex</span> <span class="kw">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="no">complexes_query_genes_join</span>$<span class="no">components_genesymbols</span>, <span class="st">"_"</span>))

<span class="co">## We can perform an enrichment analyses with the genes in the complex</span>
<span class="no">EnrichmentResults</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html">gost</a></span>(<span class="no">genes_complex</span>, <span class="kw">significant</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
    <span class="kw">user_threshold</span> <span class="kw">=</span> <span class="fl">0.001</span>, <span class="kw">correction_method</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"fdr"</span>),
    <span class="kw">sources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"GO:BP"</span>,<span class="st">"GO:CC"</span>,<span class="st">"GO:MF"</span>))

<span class="co">## We show the most significant results</span>
<span class="no">EnrichmentResults</span>$<span class="no">result</span> <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">term_id</span>, <span class="no">source</span>, <span class="no">term_name</span>,<span class="no">p_value</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span>(<span class="fl">5</span>,-<span class="no">p_value</span>)</pre></body></html></div>
<pre><code>##      term_id source                      term_name      p_value
## 1 GO:0010332  GO:BP    response to gamma radiation 5.195056e-08
## 2 GO:0032392  GO:BP           DNA geometric change 3.259476e-07
## 3 GO:0032508  GO:BP           DNA duplex unwinding 3.259476e-07
## 4 GO:0010212  GO:BP response to ionizing radiation 6.422873e-07
## 5 GO:0000781  GO:CC   chromosome, telomeric region 2.960715e-07</code></pre>
</div>
<div id="annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#annotations" class="anchor"></a>Annotations</h2>
<p>Biological annotations are statements, usually traceable and curated, about the different features of a biological entity. At the genetic level, annotations describe the biological function, the subcellular situation, the DNA location and many other related properties of a particular gene or its gene products.</p>
<p>The annotations query provides a large variety of data about proteins and complexes. These data come from dozens of databases and each kind of annotation record contains different fields. Because of this, here we have a record_id field which is unique within the records of each database. Each row contains one key value pair and you need to use the record_id to connect the related key-value pairs (see examples below).</p>
<p>Now, we focus in the annotations of the complex studied in the previous section. We first inspect the different available databases in the omnipath webserver. Then, we download the annotations for our complex itself as a biological entity. We find annotations related to the nucleus and transcriptional control, which is in agreement with the enrichment analysis results of its individual components.</p>
<div class="sourceCode" id="cb129"><html><body><pre class="r"><span class="co">## We check the different annotation databases</span>
<span class="fu"><a href="../reference/get_annotation_resources.html">get_annotation_resources</a></span>()</pre></body></html></div>
<pre><code>##  [1] "Adhesome"             "Almen2009"            "Baccin2019"          
##  [4] "CancerGeneCensus"     "CancerSEA"            "CellCellInteractions"
##  [7] "CellPhoneDB"          "CellPhoneDB_complex"  "ComPPI"              
## [10] "CORUM_Funcat"         "CORUM_GO"             "CPAD"                
## [13] "CSPA"                 "CSPA_celltype"        "DGIdb"               
## [16] "DisGeNet"             "EMBRACE"              "Exocarta"            
## [19] "GO_Intercell"         "GPCRdb"               "Guide2Pharma"        
## [22] "HGNC"                 "HPA_secretome"        "HPA_subcellular"     
## [25] "HPA_tissue"           "HPMR"                 "HPMR_complex"        
## [28] "ICELLNET"             "ICELLNET_complex"     "Integrins"           
## [31] "IntOGen"              "iTALK"                "KEGG"                
## [34] "kinase.com"           "Kirouac2010"          "LOCATE"              
## [37] "LRdb"                 "Matrisome"            "MatrixDB"            
## [40] "MCAM"                 "Membranome"           "MSigDB"              
## [43] "NetPath"              "OPM"                  "Phobius"             
## [46] "Phosphatome"          "Ramilowski_location"  "Ramilowski2015"      
## [49] "SignaLink_function"   "SignaLink_pathway"    "SIGNOR"              
## [52] "Surfaceome"           "TCDB"                 "TFcensus"            
## [55] "TopDB"                "UniProt_family"       "UniProt_keyword"     
## [58] "UniProt_location"     "UniProt_tissue"       "UniProt_topology"    
## [61] "Vesiclepedia"         "Zhong2015"</code></pre>
<div class="sourceCode" id="cb131"><html><body><pre class="r"><span class="co">## We can further investigate the features of the complex selected</span>
<span class="co">## in the previous section.</span>

<span class="co">## We first get the annotations of the complex itself:</span>
<span class="no">annotations</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span>(<span class="kw">proteins</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"COMPLEX:"</span>,
  <span class="no">complexes_query_genes_join</span>$<span class="no">components_genesymbols</span>))</pre></body></html></div>
<pre><code>## Downloaded 42 annotation records.</code></pre>
<div class="sourceCode" id="cb133"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">annotations</span>,<span class="no">source</span>,<span class="no">label</span>,<span class="no">value</span>),<span class="fl">10</span>)</pre></body></html></div>
<pre><code>##    source      label                                   value
## 1  LOCATE   location                                 nucleus
## 2  LOCATE   location                               nucleolus
## 3  MSigDB collection                       reactome_pathways
## 4  MSigDB    geneset REACTOME_DNA_DOUBLE_STRAND_BREAK_REPAIR
## 5  MSigDB collection      chemical_and_genetic_perturbations
## 6  MSigDB    geneset              KAUFFMANN_DNA_REPAIR_GENES
## 7  MSigDB collection                       reactome_pathways
## 8  MSigDB    geneset                     REACTOME_DNA_REPAIR
## 9  MSigDB collection      chemical_and_genetic_perturbations
## 10 MSigDB    geneset                PUJANA_CHEK2_PCC_NETWORK</code></pre>
<p>Afterwards, we explore the annotations of the individual components of the complex in some databases. We check the pathways where these proteins are involved. Once again, we also find many nucleus related annotations when checking their cellular location.</p>
<div class="sourceCode" id="cb135"><html><body><pre class="r"><span class="co">## Then, we explore some annotations of its individual components</span>

<span class="co">## Pathways where the proteins belong:</span>
<span class="no">annotations</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span>(<span class="kw">proteins</span><span class="kw">=</span><span class="no">genes_complex</span>,
    <span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"NetPath"</span>))</pre></body></html></div>
<pre><code>## Downloaded 7 annotation records.</code></pre>
<div class="sourceCode" id="cb137"><html><body><pre class="r"><span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">annotations</span>,<span class="no">genesymbol</span>,<span class="no">value</span>)</pre></body></html></div>
<pre><code>##   genesymbol                                         value
## 1      PARP1 TNF-related weak inducer of apoptosis (TWEAK)
## 2      PARP1                            Oncostatin-M (OSM)
## 3      PARP1         Corticotropin-releasing hormone (CRH)
## 4      PARP1                        Androgen receptor (AR)
## 5      PARP1             Tumor necrosis factor (TNF) alpha
## 6      XRCC5                        Androgen receptor (AR)
## 7      XRCC6                        Androgen receptor (AR)</code></pre>
<div class="sourceCode" id="cb139"><html><body><pre class="r"><span class="co">## Cellular localization of our proteins</span>
<span class="no">annotations</span> <span class="kw">&lt;-</span><span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span>(<span class="kw">proteins</span><span class="kw">=</span><span class="no">genes_complex</span>,
   <span class="kw">resources</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ComPPI"</span>))</pre></body></html></div>
<pre><code>## Downloaded 28 annotation records.</code></pre>
<div class="sourceCode" id="cb141"><html><body><pre class="r"><span class="co">## Since we have same record_id for some results of our query, we spread</span>
<span class="co">## these records across columns</span>
<span class="fu"><a href="https://tidyr.tidyverse.org/reference/spread.html">spread</a></span>(<span class="no">annotations</span>, <span class="no">label</span>, <span class="no">value</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="fu">desc</span>(<span class="no">score</span>)) <span class="kw">%&gt;%</span>
    <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span>(<span class="fl">10</span>, <span class="no">score</span>)</pre></body></html></div>
<pre><code>##    uniprot genesymbol entity_type source record_id      location
## 1   P12956      XRCC6     protein ComPPI      2971       nucleus
## 2   P09874      PARP1     protein ComPPI     11260       nucleus
## 3   Q14191        WRN     protein ComPPI     16123       nucleus
## 4   P13010      XRCC5     protein ComPPI     13394       nucleus
## 5   P13010      XRCC5     protein ComPPI     13396      membrane
## 6   P12956      XRCC6     protein ComPPI      2975       cytosol
## 7   P13010      XRCC5     protein ComPPI     13395       cytosol
## 8   Q14191        WRN     protein ComPPI     16122       cytosol
## 9   P12956      XRCC6     protein ComPPI      2972      membrane
## 10  P12956      XRCC6     protein ComPPI      2974 extracellular
## 11  P13010      XRCC5     protein ComPPI     13393 extracellular
##                 score
## 1    0.99999997629184
## 2      0.999999887104
## 3        0.9999996544
## 4       0.99999868288
## 5               0.972
## 6               0.958
## 7               0.958
## 8                0.94
## 9  0.8600000000000001
## 10 0.8600000000000001
## 11 0.8600000000000001</code></pre>
</div>
<div id="intercell" class="section level2">
<h2 class="hasAnchor">
<a href="#intercell" class="anchor"></a>Intercell</h2>
<p>Cells perceive cues from their microenvironment and neighboring cells, and respond accordingly to ensure proper activities and coordination between them. The ensemble of these communication process is called inter-cellular signaling (<strong>intercell</strong>).</p>
<p><strong>Intercell</strong> query provides information about the roles of proteins in inter-cellular signaling (e.g. if a protein is a ligand, a receptor, an extracellular matrix (ECM) component, etc.) This query type is very similar to annotations. However, <strong>intercell</strong> data does not come from original sources, but combined from several databases by us into categories (we also refer to the original sources).</p>
<p>We first inspect the different categories available in the Omnipath webserver. Then, we focus again in our previously selected complex and we check its the location of its individual components in the inter-cellular context. We can however see that the components of this particular complex are intracellular.</p>
<div class="sourceCode" id="cb143"><html><body><pre class="r"><span class="co">## We check some of the different intercell categories</span>
<span class="fu"><a href="../reference/get_intercell_generic_categories.html">get_intercell_generic_categories</a></span>()</pre></body></html></div>
<pre><code>## Downloaded 1143 records.</code></pre>
<pre><code>##  [1] "plasma_membrane"                     "transmembrane"                      
##  [3] "peripheral"                          "transmembrane_predicted"            
##  [5] "receptor"                            "ligand"                             
##  [7] "cell_surface_ligand"                 "ecm"                                
##  [9] "secreted"                            "ion_channel"                        
## [11] "cell_surface"                        "ligand_regulator"                   
## [13] "receptor_regulator"                  "plasma_membrane_transmembrane"      
## [15] "extracellular"                       "secreted_receptor"                  
## [17] "matrix_adhesion"                     "cell_adhesion"                      
## [19] "plasma_membrane_peripheral"          "plasma_membrane_regulator"          
## [21] "secreted_enzyme"                     "cell_surface_enzyme"                
## [23] "matrix_adhesion_regulator"           "secreted_peptidase"                 
## [25] "adhesion"                            "transporter"                        
## [27] "intracellular"                       "cell_surface_peptidase"             
## [29] "intracellular_intercellular_related" "adherens_junction"                  
## [31] "ion_channel_regulator"               "extracellular_peptidase"            
## [33] "tight_junction"                      "secreted_peptidase_inhibitor"       
## [35] "secreted_enyzme"                     "desmosome"                          
## [37] "ecm_regulator"                       "gap_junction"                       
## [39] "sparc_ecm_regulator"</code></pre>
<div class="sourceCode" id="cb146"><html><body><pre class="r"><span class="co">## We import the intercell data into a dataframe</span>
<span class="no">intercell</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_intercell.html">import_omnipath_intercell</a></span>(<span class="kw">scope</span> <span class="kw">=</span> <span class="st">'generic'</span>,
    <span class="kw">aspect</span> <span class="kw">=</span> <span class="st">'locational'</span>)</pre></body></html></div>
<pre><code>## Downloaded 212225 intercellular communication role records.</code></pre>
<div class="sourceCode" id="cb148"><html><body><pre class="r"><span class="co">## We check the intercell annotations for the individual components of</span>
<span class="co">## our previous complex. We filter our data to print it in a good format</span>
<span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">intercell</span>,<span class="no">genesymbol</span> <span class="kw">%in%</span> <span class="no">genes_complex</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>(<span class="no">genesymbol</span>, <span class="no">parent</span>, <span class="kw">.keep_all</span> <span class="kw">=</span> <span class="fl">TRUE</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">category</span>, <span class="no">genesymbol</span>, <span class="no">parent</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="no">genesymbol</span>)</pre></body></html></div>
<pre><code>##        category genesymbol        parent
## 1 intracellular      PARP1 intracellular
## 2 intracellular        WRN intracellular
## 3 intracellular      XRCC5 intracellular
## 4 intracellular      XRCC6 intracellular</code></pre>
<div class="sourceCode" id="cb150"><html><body><pre class="r"><span class="co">## We close graphical connections</span>
<span class="kw">while</span> (!<span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span>(<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.list</a></span>()))  <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</pre></body></html></div>
</div>
</div>
<div id="conclusion" class="section level1">
<h1 class="hasAnchor">
<a href="#conclusion" class="anchor"></a>Conclusion</h1>
<p><em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> provides access to the wealth of data stored in the Omnipath webservice <a href="http://omnipathdb.org/" class="uri">http://omnipathdb.org/</a> from the <em>R</em> enviroment. In addition, it contains some utility functions for visualization, filtering and analysis. The main strength of <em><a href="https://bioconductor.org/packages/3.11/OmnipathR">OmnipathR</a></em> is the straightforward transformation of the different Omnipath data into commonly used <em>R</em> objects, such as dataframes and graphs. Consequently, it allows an easy integration of the different types of data and a gateway to the vast number of <em>R</em> packages dedicated to the analysis and representaiton of biological data. We highlighted these abilities in some of the examples detailed in previous sections of this document.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] gprofiler2_0.1.9 dnet_1.1.7       supraHex_1.26.0  hexbin_1.28.1   
##  [5] tidyr_1.1.0      OmnipathR_1.3.3  jsonlite_1.7.0   igraph_1.2.5    
##  [9] knitr_1.29       BiocStyle_2.16.0
## 
## loaded via a namespace (and not attached):
##  [1] tidyselect_1.1.0    xfun_0.16           purrr_0.3.4        
##  [4] lattice_0.20-41     colorspace_1.4-1    vctrs_0.3.2        
##  [7] generics_0.0.2      viridisLite_0.3.0   htmltools_0.5.0    
## [10] stats4_4.0.2        yaml_2.2.1          plotly_4.9.2.1     
## [13] rlang_0.4.7         pkgdown_1.5.1       pillar_1.4.6       
## [16] glue_1.4.1          Rgraphviz_2.32.0    BiocGenerics_0.34.0
## [19] lifecycle_0.2.0     stringr_1.4.0       munsell_0.5.0      
## [22] gtable_0.3.0        htmlwidgets_1.5.1   memoise_1.1.0      
## [25] evaluate_0.14       curl_4.3            parallel_4.0.2     
## [28] highr_0.8           Rcpp_1.0.5          backports_1.1.8    
## [31] scales_1.1.1        BiocManager_1.30.10 desc_1.2.0         
## [34] graph_1.66.0        fs_1.4.2            ggplot2_3.3.2      
## [37] digest_0.6.25       stringi_1.4.6       bookdown_0.20      
## [40] dplyr_1.0.0         rprojroot_1.3-2     grid_4.0.2         
## [43] bitops_1.0-6        tools_4.0.2         magrittr_1.5       
## [46] RCurl_1.98-1.2      lazyeval_0.2.2      tibble_3.0.3       
## [49] crayon_1.3.4        ape_5.4             pkgconfig_2.0.3    
## [52] MASS_7.3-51.6       ellipsis_0.3.1      Matrix_1.2-18      
## [55] data.table_1.13.0   httr_1.4.2          assertthat_0.2.1   
## [58] rmarkdown_2.3       rstudioapi_0.11     R6_2.4.1           
## [61] nlme_3.1-147        compiler_4.0.2</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-BenShlomo2003">
<p>Ben-Shlomo, I., S. Yu Hsu, R. Rauch, H. W. Kowalski, and A. J. W. Hsueh. 2003. “Signaling Receptome: A Genomic and Evolutionary Perspective of Plasma Membrane Receptors Involved in Signal Transduction.” <em>Science Signaling</em> 2003 (187): re9–re9. <a href="https://doi.org/10.1126/stke.2003.187.re9">https://doi.org/10.1126/stke.2003.187.re9</a>.</p>
</div>
<div id="ref-Berggrd2007">
<p>Berggård, Tord, Sara Linse, and Peter James. 2007. “Methods for the Detection and Analysis of Proteinprotein Interactions.” <em>PROTEOMICS</em> 7 (16): 2833–42. <a href="https://doi.org/10.1002/pmic.200700131">https://doi.org/10.1002/pmic.200700131</a>.</p>
</div>
<div id="ref-Dinkel2010">
<p>Dinkel, H., C. Chica, A. Via, C. M. Gould, L. J. Jensen, T. J. Gibson, and F. Diella. 2010. “Phospho.ELM: A Database of Phosphorylation Sites–Update 2011.” <em>Nucleic Acids Research</em> 39 (Database): D261–D267. <a href="https://doi.org/10.1093/nar/gkq1104">https://doi.org/10.1093/nar/gkq1104</a>.</p>
</div>
<div id="ref-Dobson2014">
<p>Dobson, László, Tamás Langó, István Reményi, and Gábor E. Tusnády. 2014. “Expediting Topology Data Gathering for the TOPDB Database.” <em>Nucleic Acids Research</em> 43 (D1): D283–D289. <a href="https://doi.org/10.1093/nar/gku1119">https://doi.org/10.1093/nar/gku1119</a>.</p>
</div>
<div id="ref-Drew2017">
<p>Drew, Kevin, Chanjae Lee, Ryan L Huizar, Fan Tu, Blake Borgeson, Claire D McWhite, Yun Ma, John B Wallingford, and Edward M Marcotte. 2017. “Integration of over 9, 000 Mass Spectrometry Experiments Builds a Global Map of Human Protein Complexes.” <em>Molecular Systems Biology</em> 13 (6): 932. <a href="https://doi.org/10.15252/msb.20167490">https://doi.org/10.15252/msb.20167490</a>.</p>
</div>
<div id="ref-GarciaAlonso2019">
<p>Garcia-Alonso, Luz, Christian H. Holland, Mahmoud M. Ibrahim, Denes Turei, and Julio Saez-Rodriguez. 2019. “Benchmark and Integration of Resources for the Estimation of Human Transcription Factor Activities.” <em>Genome Research</em> 29 (8): 1363–75. <a href="https://doi.org/10.1101/gr.240663.118">https://doi.org/10.1101/gr.240663.118</a>.</p>
</div>
<div id="ref-Giurgiu2018">
<p>Giurgiu, Madalina, Julian Reinhard, Barbara Brauner, Irmtraud Dunger-Kaltenbach, Gisela Fobo, Goar Frishman, Corinna Montrone, and Andreas Ruepp. 2018. “CORUM: The Comprehensive Resource of Mammalian Protein Complexes2019.” <em>Nucleic Acids Research</em> 47 (D1): D559–D563. <a href="https://doi.org/10.1093/nar/gky973">https://doi.org/10.1093/nar/gky973</a>.</p>
</div>
<div id="ref-Hornbeck2014">
<p>Hornbeck, Peter V., Bin Zhang, Beth Murray, Jon M. Kornhauser, Vaughan Latham, and Elzbieta Skrzypek. 2014. “PhosphoSitePlus, 2014: Mutations, PTMs and Recalibrations.” <em>Nucleic Acids Research</em> 43 (D1): D512–D520. <a href="https://doi.org/10.1093/nar/gku1267">https://doi.org/10.1093/nar/gku1267</a>.</p>
</div>
<div id="ref-Keerthikumar2016">
<p>Keerthikumar, Shivakumar, David Chisanga, Dinuka Ariyaratne, Haidar Al Saffar, Sushma Anand, Kening Zhao, Monisha Samuel, et al. 2016. “ExoCarta: A Web-Based Compendium of Exosomal Cargo.” <em>Journal of Molecular Biology</em> 428 (4): 688–92. <a href="https://doi.org/10.1016/j.jmb.2015.09.019">https://doi.org/10.1016/j.jmb.2015.09.019</a>.</p>
</div>
<div id="ref-Turei2016">
<p>Türei, Dénes, Tamás Korcsmáros, and Julio Saez-Rodriguez. 2016. “OmniPath: Guidelines and Gateway for Literature-Curated Signaling Pathway Resources.” <em>Nature Methods</em> 13 (12): 966–67. <a href="https://doi.org/10.1038/nmeth.4077">https://doi.org/10.1038/nmeth.4077</a>.</p>
</div>
<div id="ref-VentoTormo2018">
<p>Vento-Tormo, Roser, Mirjana Efremova, Rachel A. Botting, Margherita Y. Turco, Miquel Vento-Tormo, Kerstin B. Meyer, Jong-Eun Park, et al. 2018. “Single-Cell Reconstruction of the Early Maternalfetal Interface in Humans.” <em>Nature</em> 563 (7731): 347–53. <a href="https://doi.org/10.1038/s41586-018-0698-6">https://doi.org/10.1038/s41586-018-0698-6</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alberto Valdeolivas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
