<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='GO slims are subsets of the full GO which "give a broad overview of the
ontology content without the detail of the specific fine grained terms".
In order to annotate genes with GO slim terms, we take the annotations
and search all ancestors of the terms up to the root of the ontology
tree. From the ancestors we select the terms which are part of the slim
subset.'><title>GO slim gene annotations — go_annot_slim • OmnipathR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="../deps/bs3compat-0.2.5.1/tabs.js"></script><script src="../deps/bs3compat-0.2.5.1/bs3compat.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="GO slim gene annotations — go_annot_slim"><meta property="og:description" content='GO slims are subsets of the full GO which "give a broad overview of the
ontology content without the detail of the specific fine grained terms".
In order to annotate genes with GO slim terms, we take the annotations
and search all ancestors of the terms up to the root of the ontology
tree. From the ancestors we select the terms which are part of the slim
subset.'><meta property="og:image" content="https://saezlab.github.io/OmnipathR/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-10"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-10');
</script></head><body data-spy="scroll" data-target="#toc" data-headroom>
    

      <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    <a class="navbar-brand mb-0 h1" href="../index.html">OmnipathR</a>

    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <button type="button" id="version-badge" class="badge badge-default d-none d-lg-block" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Released version">3.3.2</button>

    <div id="navbar" class="collapse navbar-collapse">
      <ul class="navbar-nav mr-auto ml-3"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bioc_workshop.html">OmniPath Bioconductor workshop</a>
    <a class="dropdown-item" href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    <a class="dropdown-item" href="../articles/gene_ontology.html">Gene Ontology in OmnipathR</a>
    <a class="dropdown-item" href="../articles/nichenet.html">Using NicheNet with OmnipathR</a>
    <a class="dropdown-item" href="../articles/omnipath_intro.html">OmnipathR: an R client for the OmniPath web service</a>
    <a class="dropdown-item" href="../articles/paths.html">Pathway construction</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control mr-sm-2" aria-label="Search" name="search-input" data-search-index="/search.json" id="search-input" placeholder="Search..." autocomplete="off"></form>

      <ul class="navbar-nav ml-1"><li class="nav-item">
  <a class="nav-link" href="https://github.com/saezlab/OmnipathR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="https://twitter.com/omnipathdb">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="https://omnipathdb.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <div class="col-md-9 contents">
    <img src="../package-logo.png" class="pkg-logo" alt=""><div class="page-header pb-2 mt-4 mb-2 border-bottom toc-ignore">
    <h1 class="display-4">GO slim gene annotations</h1>
    <small class="dont-index">Source: <a href="https://github.com/saezlab/OmnipathR/blob/master/R/go.R"><code>R/go.R</code></a></small>
    <div class="d-none name"><code>go_annot_slim.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>GO slims are subsets of the full GO which "give a broad overview of the
ontology content without the detail of the specific fine grained terms".
In order to annotate genes with GO slim terms, we take the annotations
and search all ancestors of the terms up to the root of the ontology
tree. From the ancestors we select the terms which are part of the slim
subset.</p>
    </div>

    <div id="ref-usage" class="ref-usage section level2 sourceCode">
    <pre class="sourceCode r"><code><span class="fu">go_annot_slim</span><span class="op">(</span>
  organism <span class="op">=</span> <span class="st">"human"</span>,
  slim <span class="op">=</span> <span class="st">"generic"</span>,
  aspects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"C"</span>, <span class="st">"F"</span>, <span class="st">"P"</span><span class="op">)</span>,
  cache <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre>
    </div>

    <h2>Arguments</h2>
    <dl class="row" id="ref-arguments"><dt class="col-sm-3 section level2" id="pkg-arg-organism">organism</dt>
      <dd class="col-sm-9" id="ref-arguments desc"><p>Character: either "chicken", "cow", "dog", "human", "pig"
or "uniprot_all".</p></dd>
    <dt class="col-sm-3 section level2" id="pkg-arg-slim">slim</dt>
      <dd class="col-sm-9" id="ref-arguments desc"><p>Character: the GO subset (GO slim) name. Available GO
slims are: "agr" (Alliance for Genomics Resources), "generic",
"aspergillus", "candida", "drosophila", "chembl", "metagenomic",
"mouse", "plant", "pir" (Protein Information Resource), "pombe"
and "yeast".</p></dd>
    <dt class="col-sm-3 section level2" id="pkg-arg-aspects">aspects</dt>
      <dd class="col-sm-9" id="ref-arguments desc"><p>Character vector with some of the following elements:
"C" (cellular component), "F" (molecular function) and "P" (biological
process). Gene Ontology is three separate ontologies called as three
aspects. By this parameter you can control which aspects to include
in the output.</p></dd>
    <dt class="col-sm-3 section level2" id="pkg-arg-cache">cache</dt>
      <dd class="col-sm-9" id="ref-arguments desc"><p>Logical: Load the result from cache if available.</p></dd>
    </dl><div class="ref-section section level2" id="section-value">
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value" aria-hidden="true"></a>Value</h2>

    <p>A tibble (data frame) of genes annotated with ontology terms in
    in the GO slim (subset).</p>
    </div>
    <div class="ref-section section level2" id="section-details">
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details" aria-hidden="true"></a>Details</h2>

    <p>Building the GO slim is resource intensive in its current implementation.
For human annotation and generic GO slim it might take around 20 minutes.
The result is saved into the cache so next time loading the data from
there is really quick. If the <code>cache</code> option is <code>FALSE</code> the
data will be built fresh (the annotation and ontology files still might
come from cache), and the newly build GO slim will overwrite the cache
instance.</p>
    </div>
    <div class="ref-section section level2" id="section-see-also">
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also" aria-hidden="true"></a>See also</h2>

    <div class="dont-index">
<ul><li><p><code><a href="go_annot_download.html">go_annot_download</a></code></p></li>
<li><p><code><a href="go_ontology_download.html">go_ontology_download</a></code></p></li>
<li><p><code><a href="get_db.html">get_db</a></code></p></li>
</ul></div>
    </div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples" aria-hidden="true"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="va">goslim</span> <span class="op">&lt;-</span> <span class="fu">go_annot_slim</span><span class="op">(</span>organism <span class="op">=</span> <span class="st">'human'</span>, slim <span class="op">=</span> <span class="st">'generic'</span><span class="op">)</span></span>
<span class="r-in"><span class="va">goslim</span></span>
<span class="r-in"><span class="co"># # A tibble: 276,371 x 8</span></span>
<span class="r-in"><span class="co">#    db     db_object_id db_object_symbol go_id aspect db_object_name</span></span>
<span class="r-in"><span class="co">#    &lt;fct&gt;  &lt;chr&gt;        &lt;chr&gt;            &lt;chr&gt; &lt;fct&gt;  &lt;chr&gt;</span></span>
<span class="r-in"><span class="co">#  1 UniPr. A0A024RBG1   NUDT4B           GO:0. F      Diphosphoinosito.</span></span>
<span class="r-in"><span class="co">#  2 UniPr. A0A024RBG1   NUDT4B           GO:0. F      Diphosphoinosito.</span></span>
<span class="r-in"><span class="co">#  3 UniPr. A0A024RBG1   NUDT4B           GO:0. C      Diphosphoinosito.</span></span>
<span class="r-in"><span class="co">#  4 UniPr. A0A024RBG1   NUDT4B           GO:0. C      Diphosphoinosito.</span></span>
<span class="r-in"><span class="co">#  5 UniPr. A0A024RBG1   NUDT4B           GO:0. C      Diphosphoinosito.</span></span>
<span class="r-in"><span class="co"># # . with 276,366 more rows, and 2 more variables:</span></span>
<span class="r-in"><span class="co"># #   db_object_synonym &lt;chr&gt;, db_object_type &lt;fct&gt;</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-in"></span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="sticky-top">
    <nav id="toc"><h2 data-toc-skip>Contents</h2>

    </nav></div>
  </div>
</div>


      <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Alberto Valdeolivas, Denes Turei, Attila Gabor.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer></div>

  

  

  </body></html>

