<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='ReMap (http://remap.univ-amu.fr/) is a database of ChIP-Seq
experiments. It provides raw and merged peaks and CRMs (cis regulatory
motifs) with their associations to regulators (TFs). TF-target
relationships can be derived as it is written in Garcia-Alonso et al.
2019: "For ChIP-seq, we downloaded the binding peaks from ReMap and scored
the interactions between each TF and each gene according to the distance
between the TFBSs and the genes’ transcription start sites. We evaluated
different filtering strategies that consisted of selecting only the
top-scoring 100, 200, 500, and 1000 target genes for each TF."
(https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6673718/#s1title).
This function retrieves the full processed TF-target list from the data
deposited in https://zenodo.org/record/3713238.'><title>Downloads TF-target interactions from ReMap — remap_tf_target_download • OmnipathR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="../deps/bs3compat-0.2.5.1/tabs.js"></script><script src="../deps/bs3compat-0.2.5.1/bs3compat.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Downloads TF-target interactions from ReMap — remap_tf_target_download"><meta property="og:description" content='ReMap (http://remap.univ-amu.fr/) is a database of ChIP-Seq
experiments. It provides raw and merged peaks and CRMs (cis regulatory
motifs) with their associations to regulators (TFs). TF-target
relationships can be derived as it is written in Garcia-Alonso et al.
2019: "For ChIP-seq, we downloaded the binding peaks from ReMap and scored
the interactions between each TF and each gene according to the distance
between the TFBSs and the genes’ transcription start sites. We evaluated
different filtering strategies that consisted of selecting only the
top-scoring 100, 200, 500, and 1000 target genes for each TF."
(https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6673718/#s1title).
This function retrieves the full processed TF-target list from the data
deposited in https://zenodo.org/record/3713238.'><meta property="og:image" content="https://saezlab.github.io/OmnipathR/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-10"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-10');
</script></head><body data-spy="scroll" data-target="#toc" data-headroom>
    

      <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    <a class="navbar-brand mb-0 h1" href="../index.html">OmnipathR</a>

    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <button type="button" id="version-badge" class="badge badge-default d-none d-lg-block" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Released version">3.3.2</button>

    <div id="navbar" class="collapse navbar-collapse">
      <ul class="navbar-nav mr-auto ml-3"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bioc_workshop.html">OmniPath Bioconductor workshop</a>
    <a class="dropdown-item" href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    <a class="dropdown-item" href="../articles/gene_ontology.html">Gene Ontology in OmnipathR</a>
    <a class="dropdown-item" href="../articles/nichenet.html">Using NicheNet with OmnipathR</a>
    <a class="dropdown-item" href="../articles/omnipath_intro.html">OmnipathR: an R client for the OmniPath web service</a>
    <a class="dropdown-item" href="../articles/paths.html">Pathway construction</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control mr-sm-2" aria-label="Search" name="search-input" data-search-index="/search.json" id="search-input" placeholder="Search..." autocomplete="off"></form>

      <ul class="navbar-nav ml-1"><li class="nav-item">
  <a class="nav-link" href="https://github.com/saezlab/OmnipathR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="https://twitter.com/omnipathdb">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="https://omnipathdb.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <div class="col-md-9 contents">
    <img src="../package-logo.png" class="pkg-logo" alt=""><div class="page-header pb-2 mt-4 mb-2 border-bottom toc-ignore">
    <h1 class="display-4">Downloads TF-target interactions from ReMap</h1>
    <small class="dont-index">Source: <a href="https://github.com/saezlab/OmnipathR/blob/master/R/remap.R"><code>R/remap.R</code></a></small>
    <div class="d-none name"><code>remap_tf_target_download.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>ReMap (<a href="http://remap.univ-amu.fr/">http://remap.univ-amu.fr/</a>) is a database of ChIP-Seq
experiments. It provides raw and merged peaks and CRMs (cis regulatory
motifs) with their associations to regulators (TFs). TF-target
relationships can be derived as it is written in Garcia-Alonso et al.
2019: "For ChIP-seq, we downloaded the binding peaks from ReMap and scored
the interactions between each TF and each gene according to the distance
between the TFBSs and the genes’ transcription start sites. We evaluated
different filtering strategies that consisted of selecting only the
top-scoring 100, 200, 500, and 1000 target genes for each TF."
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6673718/#s1title">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6673718/#s1title</a>).
This function retrieves the full processed TF-target list from the data
deposited in <a href="https://zenodo.org/record/3713238">https://zenodo.org/record/3713238</a>.</p>
    </div>

    <div id="ref-usage" class="ref-usage section level2 sourceCode">
    <pre class="sourceCode r"><code><span class="fu">remap_tf_target_download</span><span class="op">(</span><span class="op">)</span></code></pre>
    </div>


    <div class="ref-section section level2" id="section-value">
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value" aria-hidden="true"></a>Value</h2>

    <p>Data frame with TF-target relationships.</p>
    </div>
    <div class="ref-section section level2" id="section-see-also">
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also" aria-hidden="true"></a>See also</h2>

    <div class="dont-index">
<ul><li><p><code><a href="remap_dorothea_download.html">remap_dorothea_download</a></code></p></li>
<li><p><code><a href="remap_filtered.html">remap_filtered</a></code></p></li>
</ul></div>
    </div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples" aria-hidden="true"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="va">remap_interactions</span> <span class="op">&lt;-</span> <span class="fu">remap_tf_target_download</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">remap_interactions</span></span>
<span class="r-in"><span class="co"># # A tibble: 9,546,470 x 4</span></span>
<span class="r-in"><span class="co">#    source_genesymbol target_genesymbol target_ensembl     score</span></span>
<span class="r-in"><span class="co">#    &lt;chr&gt;             &lt;chr&gt;             &lt;chr&gt;              &lt;dbl&gt;</span></span>
<span class="r-in"><span class="co">#  1 ADNP              PTPRS             ENSG00000105426.16  1000</span></span>
<span class="r-in"><span class="co">#  2 AFF4              PRKCH             ENSG00000027075.14  1000</span></span>
<span class="r-in"><span class="co">#  3 AHR               CTNND2            ENSG00000169862.18  1000</span></span>
<span class="r-in"><span class="co">#  4 AR                PDE4D             ENSG00000113448.18  1000</span></span>
<span class="r-in"><span class="co">#  5 ARID1A            PLEC              ENSG00000178209.14  1000</span></span>
<span class="r-in"><span class="co"># # . with 9,546,460 more rows</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-in"></span>
<span class="r-in"></span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="sticky-top">
    <nav id="toc"><h2 data-toc-skip>Contents</h2>

    </nav></div>
  </div>
</div>


      <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Alberto Valdeolivas, Denes Turei, Attila Gabor.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer></div>

  

  

  </body></html>

