<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Translate gene and protein identifiers — translate_ids • OmnipathR</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Translate gene and protein identifiers — translate_ids" />
<meta property="og:description" content="Translates a vector of identifiers, resulting a new vector, or a column
of identifiers in a data frame by creating another column with the target
identifiers." />
<meta property="og:image" content="https://saezlab.github.io/OmnipathR/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-10"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-10');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OmnipathR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">3.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bioc_workshop.html">OmniPath Bioconductor workshop</a>
    </li>
    <li>
      <a href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    </li>
    <li>
      <a href="../articles/gene_ontology.html">Gene Ontology in OmnipathR</a>
    </li>
    <li>
      <a href="../articles/nichenet.html">Using NicheNet with OmnipathR</a>
    </li>
    <li>
      <a href="../articles/omnipath_intro.html">OmnipathR: an R client for the OmniPath web service</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/saezlab/OmnipathR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/omnipathdb">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://omnipathdb.org/">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Translate gene and protein identifiers</h1>
    <small class="dont-index">Source: <a href='https://github.com/saezlab/OmnipathR/blob/master/R/uniprot.R'><code>R/uniprot.R</code></a></small>
    <div class="hidden name"><code>translate_ids.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Translates a vector of identifiers, resulting a new vector, or a column
of identifiers in a data frame by creating another column with the target
identifiers.</p>
    </div>

    <pre class="usage"><span class='fu'>translate_ids</span><span class='op'>(</span>
  <span class='va'>d</span>,
  <span class='va'>...</span>,
  uploadlists <span class='op'>=</span> <span class='cn'>FALSE</span>,
  keep_untranslated <span class='op'>=</span> <span class='cn'>TRUE</span>,
  return_df <span class='op'>=</span> <span class='cn'>FALSE</span>,
  organism <span class='op'>=</span> <span class='fl'>9606</span>,
  reviewed <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>d</th>
      <td><p>Character vector or data frame.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>At least two arguments, with or without names. The first
of these arguments describes the source identifier, the rest
of them describe the target identifier(s). The values of all these
arguments must be valid identifier types as shown in Details. The
names of the arguments are column names. In case of the first
(source) ID the column must exist. For the rest of the IDs new
columns will be created with the desired names. For ID types provided
as arguments without names, the name of the ID type will be used for
column name.</p></td>
    </tr>
    <tr>
      <th>uploadlists</th>
      <td><p>Force using the <code>uploadlists</code> service from UniProt.
By default the plain query interface is used (implemented in
<code><a href='uniprot_full_id_mapping_table.html'>uniprot_full_id_mapping_table</a></code> in this package).
If any of the provided ID types is only available in the uploadlists
service, it will be automatically selected. The plain query interface
is preferred because in the long term, with caching, it requires
less download and data storage.</p></td>
    </tr>
    <tr>
      <th>keep_untranslated</th>
      <td><p>In case the output is a data frame, keep the
records where the source identifier could not be translated. At
these records the target identifier will be NA.</p></td>
    </tr>
    <tr>
      <th>return_df</th>
      <td><p>Return a data frame even if the input is a vector.</p></td>
    </tr>
    <tr>
      <th>organism</th>
      <td><p>Integer, NCBI Taxonomy ID of the organism (by default
9606 for human). Matters only if <code>uploadlists</code> is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>reviewed</th>
      <td><p>Translate only reviewed (<code>TRUE</code>), only unreviewed
(<code>FALSE</code>) or both (<code>NULL</code>) UniProt records. Matters only
if <code>uploadlists</code> is <code>FALSE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    
<ul>
<li><p>Data frame: if the input is a data frame or the input is a
vector and <code>return_df</code> is <code>TRUE</code>.</p></li>
<li><p>Vector: if the input is a vector, there is only one target
ID type and <code>return_df</code> is <code>FALSE</code>.</p></li>
<li><p>List of vectors: if the input is a vector, there are more than
one target ID types and <code>return_df</code> is <code>FALSE</code>. The names
of the list will be ID types (as they were column names, see
the description of the <code>...</code> argument), and the list will also
include the source IDs.</p></li>
</ul>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function, depending on the <code>uploadlists</code> parameter, uses either
the uploadlists service of UniProt or plain UniProt queries to obtain
identifier translation tables. The possible values for <code>from</code> and <code>to</code>
are the identifier type abbreviations used in the UniProt API, please
refer to the table here: <a href='https://www.uniprot.org/help/api_idmapping'>https://www.uniprot.org/help/api_idmapping</a>.
In addition, simple synonyms are available which realize a uniform API
for the uploadlists and UniProt query based backends. These are the
followings:</p><pre>| OmnipathR      | Uploadlists          | UniProt query           |
| -------------- | -------------------- | ----------------------- |
| uniprot        | ACC                  | id                      |
| uniprot_entry  | ID                   | entry name              |
| genesymbol     | GENENAME             | genes(PREFERRED)        |
| genesymbol_syn |                      | genes(ALTERNATIVE)      |
| hgnc           | HGNC_ID              | database(HGNC)          |
| entrez         | P_ENTREZGENEID       | database(geneid)        |
| ensg           | ENSEMBLGENOME_ID     |                         |
| enst           | ENSEMBL_TRS_ID       | database(ensembl)       |
| ensp           | ENSEMBL_PRO_ID       |                         |
| ensgt          | ENSEMBLGENOME_TRS_ID |                         |
| ensgp          | ENSEMBLGENOME_PRO_ID |                         |
| ensembl        | ENSEMBL_ID           |                         |
| protein_name   |                      | protein names           |
| refseqp        | P_REFSEQ_AC          | database(refseq)        |
| refseqn        | REFSEQ_NT_ID         |                         |
| embl           | EMBL                 | database(embl)          |
| embl_id        | EMBL_ID              |                         |
| gi             | P_GI                 |                         |
| pir            | PIR                  |                         |
| pdb            | PDB_ID               |                         |
</pre>

<p>The mapping between identifiers can be ambiguous. In this case one row
in the original data frame yields multiple rows or elements in the
returned data frame or vector(s).</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>
<ul>
<li><p><code><a href='uniprot_id_mapping_table.html'>uniprot_id_mapping_table</a></code></p></li>
<li><p><code><a href='uniprot_full_id_mapping_table.html'>uniprot_full_id_mapping_table</a></code></p></li>
</ul>
</div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>d</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>uniprot_id <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'P00533'</span>, <span class='st'>'Q9ULV1'</span>, <span class='st'>'P43897'</span>, <span class='st'>'Q9Y2P5'</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>d</span> <span class='op'>&lt;-</span> <span class='fu'>translate_ids</span><span class='op'>(</span><span class='va'>d</span>, uniprot_id <span class='op'>=</span> <span class='va'>uniprot</span>, <span class='va'>genesymbol</span><span class='op'>)</span>
<span class='va'>d</span>
</div><div class='output co'>#&gt;   uniprot_id genesymbol
#&gt; 1     P00533       EGFR
#&gt; 2     Q9ULV1       FZD4
#&gt; 3     P43897       TSFM
#&gt; 4     Q9Y2P5    SLC27A5</div><div class='input'><span class='co'>#   uniprot_id genesymbol</span>
<span class='co'># 1     P00533       EGFR</span>
<span class='co'># 2     Q9ULV1       FZD4</span>
<span class='co'># 3     P43897       TSFM</span>
<span class='co'># 4     Q9Y2P5    SLC27A5</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alberto Valdeolivas, Denes Turei, Attila Gabor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


